# Conclusions {.unnumbered}
Since the last decade, the rapid advances in genotyping technologies
have changed the way genes involved in mendelian disorders and complex
diseases are mapped, moving from candidate genes approaches to linkage
disequilibrium mapping, of which GWAS is a large-scale example. In the
mid-1990s, some researchers already foresaw the coming of the GWAS era
and the crucial contribution of high-throughput genotyping technologies
in the field of genetic epidemiology. Indeed, [@risch1996future] noted
that small genetic effects could be detected with greater power by
association analyses and proposed that genome-wide LD mapping (GWAS)
could be applied if technologies were developed to study SNP frequencies
in all genes, contrasting in ill cases vs. control subjects. On another
side, [@lander1996new] suggested the common disease common variant
(CDCV) hypothesis and proposed cataloguing common SNP (with MAF
$\geq 5\%$) and studying their association to disease in large samples.
GWAS strategy under the CDCV hypothesis assumed that many different
common SNP have small effects on each disease, and that some could be
found by testing enough SNP in enough people.

Since 2005 ([@klein2005complement]), GWAS have produced strongly
significant evidence that specific common DNA sequence differences among
people influence their genetic susceptibility to many different common
diseases [@manolio2008hapmap]. However, they are also subject to several
limitations intrinsic to the types of data but also to the statistical
methods used. On one side the strong correlations between genetic
variants, population structure, epistasis or effect size of rare-variant
are partly responsible for the missing heritability. But on the other
hand, although the single marker method remains the most widely used
approach in GWAS, its relevance may be called into question in the
context of complex diseases.

The new methodologies developed during this PhD are therefore part of
this context. We try with this manuscript to provide a thorough
introduction to GWAS by reminding in a first time the genetic precepts
fundamental to the understanding of our works but also by introducing
the concept of statistical learning. We chose not only to detail several
state-of-the-art methods used in GWAS but also to put a particular
emphasis on statistical learning by devoting an entire chapter to it.
This choice was motivated by the conviction that a multidisciplinary
approach combining both biological and statistical learning knowledges
can help to understand the limits of traditional methods used in GWAS
but also to imagine potential levers for improvement in terms of
methodology.

### Discussions on LEOS algorithm  {.unnumbered}

Based on the observation that baseline single-marker analysis in GWAS is
strongly affected by the multiple testing burden due to the high
dimensionality of the data leading to the inability to identify variants
having small effect on phenotype, we first came up with the idea of
aggregating SNP within a same LD block for a dimension-reduction
purpose. This reasoning led to the development of the method LEOS,
described in Chapter \@ref(LEOS). In this work we proposed a four-step
algorithm explicitly designed to take benefit of the linkage
disequilibrium structure in GWAS data. LEOS combines, on the one hand,
unsupervised learning methods that cluster correlated-SNP, and on the
other hand, supervised learning techniques that identify the optimal
number of clusters and reduce the dimension of the predictor matrix.

The evaluation of the method was carried out from both a predictive and
explanatory point of view. One part of the method consist in finding the
optimal group structure to construct a matrix on new aggregated-SNP
variables using supervised learning techniques. We noticed, in the
assessment of the method on simulated and real datasets, that the
combination of our aggregating function with a ridge regression model
leads to a major improvement in terms of predictive power when the
linkage disequilibrium structure is strong enough, hence suggesting the
existence of multivariate effects due to the combination of several SNP.
Furthermore, when using high-dimensional generalized additive model
(HGAM) in place of linear models, we remarked that we were able to
further increase the predictive accuracy. These results suggest a first
interesting feature of our method if one wants to predict a phenotype
based solely on genetic markers, with possible application in
personalized medicine. However, these preliminary results, although
encouraging, must be subjected to additional tests such as a comparative
analysis with other machine learning algorithms specialized in the
predictive aspect. It also seems important to confirm the robustness of
these results on other data sets and on replicative studies.

Although the predictive aspect of the algorithm is of crucial
importance, the main objective we had in mind while developing the
method was to find a way to increase statistical power and precision in
GWAS. Regarding this matter, accounting for the linkage disequilibrium
structure of the genome and aggregating highly-correlated SNP is seen to
be a powerful alternative to standard marker analysis. Indeed, LEOS
demonstrates its ability, in different simulation scenarios, to retrieve
true causal SNP and/or clusters of SNP with substantially higher
precision coupled with a good power than standard approaches. Even
though it has been able to recover a genomic region known to be
associated with ankylosing spondylitis, we have not been able to detect
new genomic regions significantly associated with the disease, certainly
suggesting that some effects might still be too small to be detected or
that there are other causes that cannot be detected with this type of
approach, such as effects of interactions with the environment or
epistasis. We also investigated, using HGAM on the aggregated-SNP
matrix, the possibility to detect non-linear relationship with the
phenotype. Albeit the regions identified did not differ from those
previously identified with a classical linear regression model, the
results obtained on the AS dataset still point interesting non-linear
patterns between some aggregated-SNP in the specific HLA region of
chromosome 6 and the phenotype. Nevertheless, we remain convinced that
generalized additive models could be of great benefit in GWAS,
particularly in terms of predictive power but also in the identification
of non-linear behaviour.

### Discussions on SICOMORE algorithm {.unnumbered}

One possible way to understand the expression of certain diseases is to
consider gene-environment interactions. Sensitivity to environmental
risk factors for a disease may be inherited, leading to cases where
individuals exposed to the same environment but with different genotypes
can be affected differently, resulting in different disease phenotypes.
In the context of medical genetics and epidemiology, the study of
gene-environment interactions is of great importance. Indeed, if we
estimate only the separate contributions of genes and environment to a
disease, and ignore their interactions, we will incorrectly estimate the
fraction of phenotypic variance attributable to genes, environment, and
their joint effect. Restricting analysis of environmental factors in
epidemiological studies to individuals who are genetically susceptible
to the exposure should increase the magnitude of relative risks,
increasing the confidence that the observed associations are not due to
chance [@hunter2005gene].

A possible lead to investigate gene-environment interactions is take
into account the contribution of microbial communities on the expression
of a phenotype. As previously stated, there is growing evidences of the
role of microbiome in basic biological processes whether in progression
of major human diseases or in plant growth. These facts motivated the
development of a new statistical method to tackle the detection of such
interactions in a GWAS context. This topic offers many statistical
challenges, among which the way to deal with the multiple testing
burden. That is why we choose to use the idea to compress the data, as
with the LEOS method, and to combine several statistical learning
methods to develop an algorithm dedicated to the search for statistical
interactions, with a focus on genomic and metagenomic data.

The SICOMORE method, described in Chapter \@ref(sicomore),
advantageously uses the strengths of different existing methods to
combine them in a powerful single algorithm. First of all, we
constructed the hierarchy of the genetic data with a well-proven
spatially-constrained hierarchical clustering adapted to SNP data
developed by [@dehman_performance_2015]. Secondly, taking the average
values of strongly correlated predictors, such as SNP within the same
LD-block, and use them into a predictive model has already proved by
[@park_averaged_2007] to be a powerful approach. Finally, we took
benefit of the weighting scheme proposed by [@grimonprez_selection_2016]
for the selection of the supervariables in the lasso procedure where we
used a penalty factor defined by the length of the gap in the
hierarchical tree, as explained in SectionÂ \@ref(structure).

We evaluated and compared the performance SICOMORE with others methods
in terms of power and precision. The results have put forward that, in
terms of precision, all methods exhibit weak performances mainly due to
the fact that the algorithms select groups which contain too many
variables. As for the statistical power, SICOMORE always exhibited in
the numerical simulations the strongest recall compared to the other
methods. The application of our method to the *Medicago truncatula*
dataset highlighted some significant interactions between genomic and
metagenomic features in relation with three different phenotypes.
However, although promising, these results need to be confirmed by a
relevant biological interpretation that will be carried out by a
discussion with our collaborators from INRA who have gracefully provided
us these data. This should allow to append a biological interpretation
to these results in the paper to come (currently in a preprint state).

Despite these interesting results, SICOMORE is nonetheless subject to
some limitations that need to be addressed in future works. First of
all, although the lasso procedure to select the supervariables in both
complementary datasets is relevant for a dimension-reduction purpose, it
may induce some biases in the multiple testing procedure we use
afterwards because we perform a variable selection step before adjusting
the $p$-values. One way around this problem could be to use post-hoc
inference for multiple comparisons [@goeman2011multiple].

Secondly, as observed in the analysis of the *Medicago truncatula*
dataset, the stability of the variable selection step is problematic.
The use of a variable selection model other than the lasso may
circumvent this issue, with for instance the *Bolasso* model
[@bach2008bolasso] where the author proposed to intersect the supports
of replicated bootstrapped Lasso estimates for consistent model
selection. In the same fashion, [@meinshausen2010stability] introduced
the stability selection based on subsampling in combination with
high-dimensional selection algorithms.

## Perspectives {.unnumbered}

The works presented in this thesis are the result of a reflection on
ways to improve GWAS studies through the creation of new data-driven
methodologies. Still, the possible contributions to the field of GWAS
brought by the development of new statistical methods are not limited to
those mentioned in this manuscript and can fall into a number of
categories depending on their objectives. To conclude, we will therefore
suggest some avenues of research not mentioned so far but worthwhile to
be explored in future works.

At first, we can mention methods designed to better modelled population
structure and relatedness between individuals in a sample during
association analyses such as the works on linear mixed models in
[@listgarten_improved_2012; @segura2012efficient; @kang2010variance] or
the methods for estimating and partitioning genetic (co)variance
[@finucane2015partitioning; @yang2010common].

In another fashion, methods combining classical statistical approaches
with Machine Learning are of interest for exploratory purposes as in
[@mieth2016combining] where multiple hypothesis tests are combined with
support vector machine (SVM) to increase statistical power. Similarly,
for purely predictive purposes, several machine learning methods such as
random forest [@geurst_exploiting_2014], classification-regression trees
(CRT) [@maciukiewicz2018gwas] or even Deep Learning (Neural Network)
[@fergus2018utilising] are also worthwhile considering in GWAS.

At last, the discovery of causal pathways between genomes and molecular
traits such as gene expression, DNA methylation, or metabolites is of
great importance to unravel cause and consequence in genetic
epidemiology. The combination of sequence variation with molecular
phenotypes, disease data and environmental covariates with novel
analytical methods such as Mendelian randomization
[@davey2003mendelian; @zhu2018causal] or causal Bayesian networks as in
[@rau2013joint] have great potential in this respect.
