<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>5.2 Learning with complementary datasets | Statistical learning for omics association and interactions studies based on blockwise feature compression</title>
  <meta name="description" content="PdD thesis">
  <meta name="generator" content="bookdown  and GitBook 2.6.7">

  <meta property="og:title" content="5.2 Learning with complementary datasets | Statistical learning for omics association and interactions studies based on blockwise feature compression />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="PdD thesis" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="5.2 Learning with complementary datasets | Statistical learning for omics association and interactions studies based on blockwise feature compression />
  
  <meta name="twitter:description" content="PdD thesis" />
  

<meta name="author" content="Florent Guinot">


<meta name="date" content="2019-01-11">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="introduction-3.html">
<link rel="next" href="implementation.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />










<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Abstract</a></li>
<li class="chapter" data-level="" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i>General introduction</a><ul>
<li class="chapter" data-level="" data-path="intro.html"><a href="intro.html#background"><i class="fa fa-check"></i>Background</a></li>
<li class="chapter" data-level="" data-path="intro.html"><a href="intro.html#problematic"><i class="fa fa-check"></i>Problematic</a></li>
<li class="chapter" data-level="" data-path="intro.html"><a href="intro.html#objectives"><i class="fa fa-check"></i>Objectives</a></li>
<li class="chapter" data-level="" data-path="intro.html"><a href="intro.html#contributions"><i class="fa fa-check"></i>Contributions</a></li>
<li class="chapter" data-level="" data-path="intro.html"><a href="intro.html#contents-of-the-manuscript"><i class="fa fa-check"></i>Contents of the manuscript</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="basic-concepts-of-molecular-genetics.html"><a href="basic-concepts-of-molecular-genetics.html"><i class="fa fa-check"></i><b>1</b> Basic concepts of molecular genetics</a><ul>
<li class="chapter" data-level="1.1" data-path="genome.html"><a href="genome.html"><i class="fa fa-check"></i><b>1.1</b> Genome description</a></li>
<li class="chapter" data-level="1.2" data-path="genome-sequencing.html"><a href="genome-sequencing.html"><i class="fa fa-check"></i><b>1.2</b> Genome sequencing</a><ul>
<li class="chapter" data-level="1.2.1" data-path="genome-sequencing.html"><a href="genome-sequencing.html#DNAseq"><i class="fa fa-check"></i><b>1.2.1</b> DNA sequencing</a></li>
<li class="chapter" data-level="1.2.2" data-path="genome-sequencing.html"><a href="genome-sequencing.html#sequence-assembly"><i class="fa fa-check"></i><b>1.2.2</b> Sequence assembly</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="dna-polymorphism.html"><a href="dna-polymorphism.html"><i class="fa fa-check"></i><b>1.3</b> DNA polymorphism</a><ul>
<li class="chapter" data-level="1.3.1" data-path="dna-polymorphism.html"><a href="dna-polymorphism.html#restriction-fragment-length-polymorphisms-rflp"><i class="fa fa-check"></i><b>1.3.1</b> Restriction Fragment Length Polymorphisms (RFLP)</a></li>
<li class="chapter" data-level="1.3.2" data-path="dna-polymorphism.html"><a href="dna-polymorphism.html#simple-sequence-length-polymorphisms-sslp"><i class="fa fa-check"></i><b>1.3.2</b> Simple Sequence Length Polymorphisms (SSLP)</a></li>
<li class="chapter" data-level="1.3.3" data-path="dna-polymorphism.html"><a href="dna-polymorphism.html#single-nucleotide-polymorphisms-snp"><i class="fa fa-check"></i><b>1.3.3</b> Single Nucleotide Polymorphisms (SNP)</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="linkage.html"><a href="linkage.html"><i class="fa fa-check"></i><b>1.4</b> Linkage and partial linkage for genetic mapping</a></li>
<li class="chapter" data-level="1.5" data-path="basic-concepts-in-population-genetics.html"><a href="basic-concepts-in-population-genetics.html"><i class="fa fa-check"></i><b>1.5</b> Basic concepts in population genetics</a><ul>
<li class="chapter" data-level="1.5.1" data-path="basic-concepts-in-population-genetics.html"><a href="basic-concepts-in-population-genetics.html#HWE"><i class="fa fa-check"></i><b>1.5.1</b> Hardy-Weinberg equilibrium in large population</a></li>
<li class="chapter" data-level="1.5.2" data-path="basic-concepts-in-population-genetics.html"><a href="basic-concepts-in-population-genetics.html#genetic-drift-in-small-population"><i class="fa fa-check"></i><b>1.5.2</b> Genetic drift in small population</a></li>
<li class="chapter" data-level="1.5.3" data-path="basic-concepts-in-population-genetics.html"><a href="basic-concepts-in-population-genetics.html#concept-of-heritability"><i class="fa fa-check"></i><b>1.5.3</b> Concept of heritability</a></li>
</ul></li>
<li class="chapter" data-level="1.6" data-path="LD.html"><a href="LD.html"><i class="fa fa-check"></i><b>1.6</b> Linkage disequilibrium</a><ul>
<li class="chapter" data-level="1.6.1" data-path="LD.html"><a href="LD.html#definition"><i class="fa fa-check"></i><b>1.6.1</b> Definition</a></li>
<li class="chapter" data-level="1.6.2" data-path="LD.html"><a href="LD.html#measure-of-ld"><i class="fa fa-check"></i><b>1.6.2</b> Measure of LD</a></li>
<li class="chapter" data-level="1.6.3" data-path="LD.html"><a href="LD.html#estimation-of-linkage-disequilibrium"><i class="fa fa-check"></i><b>1.6.3</b> Estimation of linkage disequilibrium</a></li>
<li class="chapter" data-level="1.6.4" data-path="LD.html"><a href="LD.html#origins-of-linkage-disequilibrium"><i class="fa fa-check"></i><b>1.6.4</b> Origins of linkage disequilibrium</a></li>
</ul></li>
<li class="chapter" data-level="1.7" data-path="haplo.html"><a href="haplo.html"><i class="fa fa-check"></i><b>1.7</b> Structure of haplotype blocks in the human genome</a><ul>
<li class="chapter" data-level="" data-path="haplo.html"><a href="haplo.html#definition-of-haplotype-blocks"><i class="fa fa-check"></i>Definition of haplotype blocks</a></li>
<li class="chapter" data-level="" data-path="haplo.html"><a href="haplo.html#patterns-in-human-genome"><i class="fa fa-check"></i>Patterns in human genome</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="statistical-context.html"><a href="statistical-context.html"><i class="fa fa-check"></i><b>2</b> Statistical context</a><ul>
<li class="chapter" data-level="2.1" data-path="notations.html"><a href="notations.html"><i class="fa fa-check"></i><b>2.1</b> Notations</a></li>
<li class="chapter" data-level="2.2" data-path="concepts-of-statistical-learning.html"><a href="concepts-of-statistical-learning.html"><i class="fa fa-check"></i><b>2.2</b> Concepts of statistical learning</a><ul>
<li class="chapter" data-level="2.2.1" data-path="concepts-of-statistical-learning.html"><a href="concepts-of-statistical-learning.html#prediction"><i class="fa fa-check"></i><b>2.2.1</b> Prediction</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="parametric.html"><a href="parametric.html"><i class="fa fa-check"></i><b>2.3</b> Parametric methods</a><ul>
<li class="chapter" data-level="2.3.1" data-path="parametric.html"><a href="parametric.html#linmod"><i class="fa fa-check"></i><b>2.3.1</b> Linear models</a></li>
<li class="chapter" data-level="2.3.2" data-path="parametric.html"><a href="parametric.html#penalized"><i class="fa fa-check"></i><b>2.3.2</b> Penalized linear regression</a></li>
<li class="chapter" data-level="2.3.3" data-path="parametric.html"><a href="parametric.html#glm"><i class="fa fa-check"></i><b>2.3.3</b> Generalized linear models</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="non-parametric.html"><a href="non-parametric.html"><i class="fa fa-check"></i><b>2.4</b> Splines and generalized additive models: Moving beyond linearit</a><ul>
<li class="chapter" data-level="2.4.1" data-path="non-parametric.html"><a href="non-parametric.html#introduction"><i class="fa fa-check"></i><b>2.4.1</b> Introduction</a></li>
<li class="chapter" data-level="2.4.2" data-path="non-parametric.html"><a href="non-parametric.html#splines"><i class="fa fa-check"></i><b>2.4.2</b> Regression splines</a></li>
<li class="chapter" data-level="2.4.3" data-path="non-parametric.html"><a href="non-parametric.html#mathrmb-splines"><i class="fa fa-check"></i><b>2.4.3</b> <span class="math inline">\(\mathrm{B}\)</span>-splines</a></li>
<li class="chapter" data-level="2.4.4" data-path="non-parametric.html"><a href="non-parametric.html#smoothing"><i class="fa fa-check"></i><b>2.4.4</b> Cubic smoothing splines</a></li>
<li class="chapter" data-level="2.4.5" data-path="non-parametric.html"><a href="non-parametric.html#gam"><i class="fa fa-check"></i><b>2.4.5</b> Generalized additive models (GAM)</a></li>
<li class="chapter" data-level="2.4.6" data-path="non-parametric.html"><a href="non-parametric.html#hgam"><i class="fa fa-check"></i><b>2.4.6</b> High-dimensional generalized additive models (HGAM)</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="combining-cluster-analysis-and-variable-selection.html"><a href="combining-cluster-analysis-and-variable-selection.html"><i class="fa fa-check"></i><b>2.5</b> Combining cluster analysis and variable selection</a><ul>
<li class="chapter" data-level="2.5.1" data-path="combining-cluster-analysis-and-variable-selection.html"><a href="combining-cluster-analysis-and-variable-selection.html#CAH"><i class="fa fa-check"></i><b>2.5.1</b> Hierarchical clustering</a></li>
<li class="chapter" data-level="2.5.2" data-path="combining-cluster-analysis-and-variable-selection.html"><a href="combining-cluster-analysis-and-variable-selection.html#HCAR"><i class="fa fa-check"></i><b>2.5.2</b> Hierarchical Clustering and Averaging Regression</a></li>
<li class="chapter" data-level="2.5.3" data-path="combining-cluster-analysis-and-variable-selection.html"><a href="combining-cluster-analysis-and-variable-selection.html#MLGL"><i class="fa fa-check"></i><b>2.5.3</b> Multi-Layer Group-Lasso (MLGL)</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="hypothesis.html"><a href="hypothesis.html"><i class="fa fa-check"></i><b>2.6</b> Statistical testing of significance</a><ul>
<li class="chapter" data-level="2.6.1" data-path="hypothesis.html"><a href="hypothesis.html#introduction-1"><i class="fa fa-check"></i><b>2.6.1</b> Introduction</a></li>
<li class="chapter" data-level="2.6.2" data-path="hypothesis.html"><a href="hypothesis.html#chi2"><i class="fa fa-check"></i><b>2.6.2</b> <span class="math inline">\(\chi^2\)</span> test</a></li>
<li class="chapter" data-level="2.6.3" data-path="hypothesis.html"><a href="hypothesis.html#LRT"><i class="fa fa-check"></i><b>2.6.3</b> Likelihood ratio test</a></li>
<li class="chapter" data-level="2.6.4" data-path="hypothesis.html"><a href="hypothesis.html#pvalGAM"><i class="fa fa-check"></i><b>2.6.4</b> Calculation of <em>p</em>-values in GAM</a></li>
<li class="chapter" data-level="2.6.5" data-path="hypothesis.html"><a href="hypothesis.html#multiple"><i class="fa fa-check"></i><b>2.6.5</b> Multiple testing comparison</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="genome-wide-association-studies.html"><a href="genome-wide-association-studies.html"><i class="fa fa-check"></i><b>3</b> Genome-Wide Association Studies</a><ul>
<li class="chapter" data-level="3.1" data-path="introduction-2.html"><a href="introduction-2.html"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="genoquality.html"><a href="genoquality.html"><i class="fa fa-check"></i><b>3.2</b> Genotype quality control</a><ul>
<li class="chapter" data-level="3.2.1" data-path="genoquality.html"><a href="genoquality.html#deviation-from-hwe."><i class="fa fa-check"></i><b>3.2.1</b> Deviation from HWE.</a></li>
<li class="chapter" data-level="3.2.2" data-path="genoquality.html"><a href="genoquality.html#missing-data."><i class="fa fa-check"></i><b>3.2.2</b> Missing data.</a></li>
<li class="chapter" data-level="3.2.3" data-path="genoquality.html"><a href="genoquality.html#distribution-of-test-statistics."><i class="fa fa-check"></i><b>3.2.3</b> Distribution of test statistics.</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="OR.html"><a href="OR.html"><i class="fa fa-check"></i><b>3.3</b> Disease penetrance and odds ratio</a></li>
<li class="chapter" data-level="3.4" data-path="SMA.html"><a href="SMA.html"><i class="fa fa-check"></i><b>3.4</b> Single Marker Analysis</a><ul>
<li class="chapter" data-level="3.4.1" data-path="SMA.html"><a href="SMA.html#pearsons-chi2-statistic"><i class="fa fa-check"></i><b>3.4.1</b> Pearson’s <span class="math inline">\(\chi^2\)</span> statistic</a></li>
<li class="chapter" data-level="3.4.2" data-path="SMA.html"><a href="SMA.html#cochran-armitage-trend-test"><i class="fa fa-check"></i><b>3.4.2</b> Cochran-Armitage trend test</a></li>
<li class="chapter" data-level="3.4.3" data-path="SMA.html"><a href="SMA.html#logitGWAS"><i class="fa fa-check"></i><b>3.4.3</b> Logistic regression and likelihood ratio test</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="GWASlimits.html"><a href="GWASlimits.html"><i class="fa fa-check"></i><b>3.5</b> Limitations</a></li>
<li class="chapter" data-level="3.6" data-path="popstructure.html"><a href="popstructure.html"><i class="fa fa-check"></i><b>3.6</b> Population structure</a><ul>
<li class="chapter" data-level="3.6.1" data-path="popstructure.html"><a href="popstructure.html#genomic-control"><i class="fa fa-check"></i><b>3.6.1</b> Genomic control</a></li>
<li class="chapter" data-level="3.6.2" data-path="popstructure.html"><a href="popstructure.html#structured-association"><i class="fa fa-check"></i><b>3.6.2</b> Structured association</a></li>
<li class="chapter" data-level="3.6.3" data-path="popstructure.html"><a href="popstructure.html#PCC"><i class="fa fa-check"></i><b>3.6.3</b> Principle components correction</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="multiloc.html"><a href="multiloc.html"><i class="fa fa-check"></i><b>3.7</b> Multi-locus analysis</a><ul>
<li class="chapter" data-level="3.7.1" data-path="multiloc.html"><a href="multiloc.html#haplotype-based-approaches"><i class="fa fa-check"></i><b>3.7.1</b> Haplotype-based approaches</a></li>
<li class="chapter" data-level="3.7.2" data-path="multiloc.html"><a href="multiloc.html#rare-variant"><i class="fa fa-check"></i><b>3.7.2</b> Rare-variant association analysis</a></li>
<li class="chapter" data-level="3.7.3" data-path="multiloc.html"><a href="multiloc.html#adjclust"><i class="fa fa-check"></i><b>3.7.3</b> LD based approach to variable selection in GWAS</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="LEOS.html"><a href="LEOS.html"><i class="fa fa-check"></i><b>4</b> Learning the Optimal in GWAS through hierarchical SNP aggregation</a><ul>
<li class="chapter" data-level="4.1" data-path="introleos.html"><a href="introleos.html"><i class="fa fa-check"></i><b>4.1</b> Related work</a></li>
<li class="chapter" data-level="4.2" data-path="leosmethod.html"><a href="leosmethod.html"><i class="fa fa-check"></i><b>4.2</b> Method</a><ul>
<li class="chapter" data-level="4.2.1" data-path="leosmethod.html"><a href="leosmethod.html#CHAC"><i class="fa fa-check"></i><b>4.2.1</b> Step 1. Constrained-HAC</a></li>
<li class="chapter" data-level="4.2.2" data-path="leosmethod.html"><a href="leosmethod.html#Dstar"><i class="fa fa-check"></i><b>4.2.2</b> Step 2. Dimension reduction function</a></li>
<li class="chapter" data-level="4.2.3" data-path="leosmethod.html"><a href="leosmethod.html#cutree"><i class="fa fa-check"></i><b>4.2.3</b> Step 3. Optimal number of groups estimation</a></li>
<li class="chapter" data-level="4.2.4" data-path="leosmethod.html"><a href="leosmethod.html#step4"><i class="fa fa-check"></i><b>4.2.4</b> Step 4. Multiple testing on aggregated-SNP variables</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="numsim.html"><a href="numsim.html"><i class="fa fa-check"></i><b>4.3</b> Numerical simulations</a><ul>
<li class="chapter" data-level="4.3.1" data-path="numsim.html"><a href="numsim.html#simupheno"><i class="fa fa-check"></i><b>4.3.1</b> Simulation of the case-control phenotype</a></li>
<li class="chapter" data-level="4.3.2" data-path="numsim.html"><a href="numsim.html#perfeval"><i class="fa fa-check"></i><b>4.3.2</b> Performance evaluation</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="results.html"><a href="results.html"><i class="fa fa-check"></i><b>4.4</b> Results</a><ul>
<li class="chapter" data-level="4.4.1" data-path="results.html"><a href="results.html#results-and-discussions-of-the-numerical-simulations"><i class="fa fa-check"></i><b>4.4.1</b> Results and discussions of the numerical simulations</a></li>
<li class="chapter" data-level="4.4.2" data-path="results.html"><a href="results.html#performance-results-for-simulated-data."><i class="fa fa-check"></i><b>4.4.2</b> Performance results for simulated data.</a></li>
<li class="chapter" data-level="4.4.3" data-path="results.html"><a href="results.html#leosappli"><i class="fa fa-check"></i><b>4.4.3</b> Application in Wellcome Trust Case Control Consortium(WTCCC) and Ankylosing Spondylitis (AS) studies</a></li>
<li class="chapter" data-level="4.4.4" data-path="results.html"><a href="results.html#realdata"><i class="fa fa-check"></i><b>4.4.4</b> Results in WTCCC and AS studies</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="leosgam.html"><a href="leosgam.html"><i class="fa fa-check"></i><b>4.5</b> Generalized additive models in GWAS</a><ul>
<li class="chapter" data-level="4.5.1" data-path="leosgam.html"><a href="leosgam.html#comparison-of-predictive-power"><i class="fa fa-check"></i><b>4.5.1</b> Comparison of predictive power</a></li>
<li class="chapter" data-level="4.5.2" data-path="leosgam.html"><a href="leosgam.html#results-of-univariate-smoothing-splines-on-aggregated-snp"><i class="fa fa-check"></i><b>4.5.2</b> Results of univariate smoothing splines on aggregated-SNP</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="discussions.html"><a href="discussions.html"><i class="fa fa-check"></i><b>4.6</b> Discussion</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="selection-of-interaction-effects-in-compressed-multiple-omics-representation.html"><a href="selection-of-interaction-effects-in-compressed-multiple-omics-representation.html"><i class="fa fa-check"></i><b>5</b> Selection of interaction effects in compressed multiple omics representation</a><ul>
<li class="chapter" data-level="5.1" data-path="introduction-3.html"><a href="introduction-3.html"><i class="fa fa-check"></i><b>5.1</b> Introduction</a><ul>
<li class="chapter" data-level="5.1.1" data-path="introduction-3.html"><a href="introduction-3.html#background-1"><i class="fa fa-check"></i><b>5.1.1</b> Background</a></li>
<li class="chapter" data-level="5.1.2" data-path="introduction-3.html"><a href="introduction-3.html#combining-genome-and-metagenome-analyses."><i class="fa fa-check"></i><b>5.1.2</b> Combining genome and metagenome analyses.</a></li>
<li class="chapter" data-level="5.1.3" data-path="introduction-3.html"><a href="introduction-3.html#taking-structures-into-account-in-association-studies."><i class="fa fa-check"></i><b>5.1.3</b> Taking structures into account in association studies.</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="modelsicomore.html"><a href="modelsicomore.html"><i class="fa fa-check"></i><b>5.2</b> Learning with complementary datasets</a><ul>
<li class="chapter" data-level="5.2.1" data-path="modelsicomore.html"><a href="modelsicomore.html#setting-and-notations"><i class="fa fa-check"></i><b>5.2.1</b> Setting and notations</a></li>
<li class="chapter" data-level="5.2.2" data-path="modelsicomore.html"><a href="modelsicomore.html#interactions-in-linear-models"><i class="fa fa-check"></i><b>5.2.2</b> Interactions in linear models</a></li>
<li class="chapter" data-level="5.2.3" data-path="modelsicomore.html"><a href="modelsicomore.html#compressdata"><i class="fa fa-check"></i><b>5.2.3</b> Compact model</a></li>
<li class="chapter" data-level="5.2.4" data-path="modelsicomore.html"><a href="modelsicomore.html#recoverinteractions"><i class="fa fa-check"></i><b>5.2.4</b> Recovering relevant interactions</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="implementation.html"><a href="implementation.html"><i class="fa fa-check"></i><b>5.3</b> Method</a><ul>
<li class="chapter" data-level="5.3.1" data-path="implementation.html"><a href="implementation.html#preprocess"><i class="fa fa-check"></i><b>5.3.1</b> Preprocessing of the data</a></li>
<li class="chapter" data-level="5.3.2" data-path="implementation.html"><a href="implementation.html#preprocessing-of-metagenomic-data"><i class="fa fa-check"></i><b>5.3.2</b> Preprocessing of metagenomic data</a></li>
<li class="chapter" data-level="5.3.3" data-path="implementation.html"><a href="implementation.html#structure"><i class="fa fa-check"></i><b>5.3.3</b> Structuring the data</a></li>
<li class="chapter" data-level="5.3.4" data-path="implementation.html"><a href="implementation.html#using-the-structure-efficiently"><i class="fa fa-check"></i><b>5.3.4</b> Using the structure efficiently</a></li>
<li class="chapter" data-level="5.3.5" data-path="implementation.html"><a href="implementation.html#identification-of-relevant-supervariables"><i class="fa fa-check"></i><b>5.3.5</b> Identification of relevant supervariables</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="XPsimu.html"><a href="XPsimu.html"><i class="fa fa-check"></i><b>5.4</b> Numerical simulations</a><ul>
<li class="chapter" data-level="5.4.1" data-path="XPsimu.html"><a href="XPsimu.html#data-generation"><i class="fa fa-check"></i><b>5.4.1</b> Data generation</a></li>
<li class="chapter" data-level="5.4.2" data-path="XPsimu.html"><a href="XPsimu.html#generation-of-the-phenotype"><i class="fa fa-check"></i><b>5.4.2</b> Generation of the phenotype</a></li>
<li class="chapter" data-level="5.4.3" data-path="XPsimu.html"><a href="XPsimu.html#comparison-of-methods"><i class="fa fa-check"></i><b>5.4.3</b> Comparison of methods</a></li>
<li class="chapter" data-level="5.4.4" data-path="XPsimu.html"><a href="XPsimu.html#evaluation-metrics"><i class="fa fa-check"></i><b>5.4.4</b> Evaluation metrics</a></li>
<li class="chapter" data-level="5.4.5" data-path="XPsimu.html"><a href="XPsimu.html#performance-results"><i class="fa fa-check"></i><b>5.4.5</b> Performance results</a></li>
<li class="chapter" data-level="5.4.6" data-path="XPsimu.html"><a href="XPsimu.html#computational-time"><i class="fa fa-check"></i><b>5.4.6</b> Computational time</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="XPINRA.html"><a href="XPINRA.html"><i class="fa fa-check"></i><b>5.5</b> Application on real data: rhizosphere of <em>Medicago truncatula</em></a><ul>
<li class="chapter" data-level="5.5.1" data-path="XPINRA.html"><a href="XPINRA.html#material"><i class="fa fa-check"></i><b>5.5.1</b> Material</a></li>
<li class="chapter" data-level="5.5.2" data-path="XPINRA.html"><a href="XPINRA.html#analysis"><i class="fa fa-check"></i><b>5.5.2</b> Analysis</a></li>
<li class="chapter" data-level="5.5.3" data-path="XPINRA.html"><a href="XPINRA.html#results-1"><i class="fa fa-check"></i><b>5.5.3</b> Results</a></li>
<li class="chapter" data-level="5.5.4" data-path="XPINRA.html"><a href="XPINRA.html#results-on-root-shoot-ratio"><i class="fa fa-check"></i><b>5.5.4</b> Results on Root Shoot Ratio</a></li>
<li class="chapter" data-level="5.5.5" data-path="XPINRA.html"><a href="XPINRA.html#results-on-specific-nitrogen-uptake"><i class="fa fa-check"></i><b>5.5.5</b> Results on Specific Nitrogen Uptake</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="discussions-1.html"><a href="discussions-1.html"><i class="fa fa-check"></i><b>5.6</b> Discussions</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="conclusions.html"><a href="conclusions.html"><i class="fa fa-check"></i>Conclusions</a><ul>
<li class="chapter" data-level="" data-path="conclusions.html"><a href="conclusions.html#discussions-on-leos-algorithm"><i class="fa fa-check"></i>Discussions on LEOS algorithm</a></li>
<li class="chapter" data-level="" data-path="conclusions.html"><a href="conclusions.html#discussions-on-sicomore-algorithm"><i class="fa fa-check"></i>Discussions on SICOMORE algorithm</a></li>
<li class="chapter" data-level="" data-path="perspectives.html"><a href="perspectives.html"><i class="fa fa-check"></i>Perspectives</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Statistical learning for omics association and interactions studies based on blockwise feature compression</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="modelsicomore" class="section level2">
<h2><span class="header-section-number">5.2</span> Learning with complementary datasets</h2>
<p>This section introduces the setting with the notations. It also sketches
the approach to define a compact model of interactions between
complementary datasets.</p>
<div id="setting-and-notations" class="section level3">
<h3><span class="header-section-number">5.2.1</span> Setting and notations</h3>
<p>Let us consider observations stemming from two complementary views,
<span class="math inline">\(\mathit{G}\)</span> (for Genomic data) and <span class="math inline">\(\mathit{M}\)</span> (for Metagenomic data), which are
gathered into a training set
<span class="math inline">\(\mathcal{T} = \{(\mathbf{x}^\mathit{G}_i, \mathbf{x}^\mathit{M}_i, y_i)\}_{i=1}^N\)</span>, where
<span class="math inline">\((\mathbf{x}^\mathit{G}_i, \mathbf{x}^\mathit{M}_i, y_i) \in \mathbb{R}^{D_\mathit{G}} \times \mathbb{R}^{D_\mathit{M}} \times \mathbb{R}\)</span>.</p>
<p>We assume an underlying biological information on <span class="math inline">\(\mathit{G}\)</span> and <span class="math inline">\(\mathit{M}\)</span>
encoded as groups. The group structure over <span class="math inline">\(\mathit{G}\)</span> is defined by
<span class="math inline">\(N_{\mathit{G}}\)</span> groups of variables
<span class="math inline">\(\mathcal{G}=\{\mathcal{G}_{g} \}_{g=1}^{N_{\mathit{G}}}\)</span>. We denote
<span class="math inline">\(\mathbf{x}_i^{g} \in \mathbb{R}^{D_{g}}\)</span>, the sample <span class="math inline">\(i\)</span> restricted to the
variables of <span class="math inline">\(\mathit{G}\)</span> from group <span class="math inline">\(\mathcal{G}_{g}\)</span>. Similarly, the group
structure over <span class="math inline">\(\mathit{M}\)</span> is defined by <span class="math inline">\(N_{\mathit{M}}\)</span> groups of variables
<span class="math inline">\(\mathcal{M}=\{\mathcal{M}_{m}\}_{m=1}^{N_{\mathit{M}}}\)</span> and
<span class="math inline">\(\mathbf{x}_i^{m} \in \mathbb{R}^{D_{m}}\)</span> is the sample <span class="math inline">\(i\)</span> restricted to the
variables of <span class="math inline">\(\mathit{M}\)</span> from group <span class="math inline">\(\mathcal{M}_{m}\)</span>.</p>
<p>We also introduce <span class="math inline">\(D_I = D_{\mathit{G}} \cdot D_{\mathit{M}}\)</span> and
<span class="math inline">\(N_I = N_{\mathit{G}} \cdot N_{\mathit{M}}\)</span>, the number of variables and the number of
groups that may interact.</p>
<p>Finally, we use the following convention: vectors of observations
indexed with <span class="math inline">\(i\)</span>, such as <span class="math inline">\(\mathbf{x}_i\)</span>, will usually be row vectors<a href="#fn15" class="footnote-ref" id="fnref15"><sup>15</sup></a> while
vectors of coefficients, such as <span class="math inline">\(\boldsymbol{\beta}\)</span>, will usually be column
vectors.</p>
</div>
<div id="interactions-in-linear-models" class="section level3">
<h3><span class="header-section-number">5.2.2</span> Interactions in linear models</h3>
<p>Interactions between data stemming from views <span class="math inline">\(\mathit{G}\)</span> and <span class="math inline">\(\mathit{M}\)</span> may be
captured in the model <span class="math display">\[\label{eq:classical_interaction_model}
  y_i =   \mathbf{x}^{\mathit{G}}_i \boldsymbol{\gamma}_{\mathit{G}}   
  + \mathbf{x}^{\mathit{M}}_i \boldsymbol{\gamma}_{\mathit{M}}  
  + \mathbf{x}^{\mathit{G}}_i \boldsymbol{\Delta}_{\mathit{G}\mathit{M}} (\mathbf{x}^{\mathit{M}}_i)^T  +
  \epsilon_i \,,\]</span> where the vectors <span class="math inline">\(\boldsymbol{\gamma}_{\mathit{G}} \in \mathbb{R}^{D_{\mathit{G}}}\)</span>
and <span class="math inline">\(\boldsymbol{\gamma}_{\mathit{M}} \in \mathbb{R}^{D_{\mathit{M}}}\)</span> respectively denote the linear
effects related to <span class="math inline">\(\mathit{G}\)</span> and <span class="math inline">\(\mathit{M}\)</span>, the matrix
<span class="math inline">\(\boldsymbol{\Delta}_{\mathit{G}\mathit{M}} \in \mathbb{R}^{D_{\mathit{G}} \times D_{\mathit{M}}}\)</span> contains the
interactions between all pairs of variables of <span class="math inline">\(\mathit{G}\)</span> and <span class="math inline">\(\mathit{M}\)</span> and
<span class="math inline">\(\epsilon_i \in \mathbb{R}\)</span> is a residual error.</p>
<p>Underlying notions in models of interactions are the one of <em>strong dependency</em> (SD) and <em>weak dependency</em> (WD), the first one being more common (see for instance <span class="citation">(Bien, Taylor, and Tibshirani <a href="#ref-bien2013lasso">2013</a>)</span> and the discussion therein).
Under the hypothesis of <em>strong dependency</em>, an interaction is effective
if and only if the corresponding single effects are also effective while
the hypothesis of <em>weak dependency</em> implies that an interaction is
effective if one of the main effect is also effective. Formally, for all
variables <span class="math inline">\(j \in \mathbf{x}^{\mathit{G}}\)</span> and for all variables <span class="math inline">\(j&#39; \in \mathbf{x}^{\mathit{M}}\)</span>, if
<span class="math inline">\(\gamma_{j}\)</span>, <span class="math inline">\(\gamma_{j&#39;}\)</span> and <span class="math inline">\(\delta_{jj&#39;}\)</span> are the coefficients
related to <span class="math inline">\(\boldsymbol{\gamma}_{\mathit{G}}\)</span>, <span class="math inline">\(\boldsymbol{\gamma}_{\mathit{M}}\)</span> and <span class="math inline">\(\boldsymbol{\Delta}_{\mathit{G}\mathit{M}}\)</span>, then
<span class="math display">\[\begin{aligned}
  &amp; (SD) &amp;&amp;  \delta_{jj&#39;} \neq 0 \qquad \Rightarrow  \qquad
            \gamma_{j} 
            \neq 0 &amp;&amp; \text{ and } &amp;&amp; \gamma_{j&#39;} \neq 0  \,, \\
  &amp;  (WD) &amp;&amp;  \delta_{jj&#39;} \neq 0 \qquad \Rightarrow  \qquad
             \gamma_{j} 
             \neq 0 &amp;&amp; \text{ or } &amp;&amp; \gamma_{j&#39;} \neq 0  \,.\end{aligned}\]</span></p>
<p>In this context, <span class="citation">(Bien, Taylor, and Tibshirani <a href="#ref-bien2013lasso">2013</a>)</span> have proposed a sparse model of
interactions which faces computational limitations for large dimensional
problems according to <span class="citation">(Lim and Hastie <a href="#ref-lim2015learning">2015</a>)</span> and <span class="citation">(She, Wang, and Jiang <a href="#ref-she2016group">2016</a>)</span>. While
<span class="citation">(Lim and Hastie <a href="#ref-lim2015learning">2015</a>)</span> introduce a method for learning pairwise interactions
in a regression model by solving a constrained overlapping Group-Lasso
<span class="citation">(Jacob, Obozinski, and Vert <a href="#ref-jacob2009group">2009</a>)</span> in a manner that satisfies strong dependencies,
<span class="citation">(She, Wang, and Jiang <a href="#ref-she2016group">2016</a>)</span> propose a formulation with an overlapping regularization
that fits both kind of hypotheses and provide theoretical insights on
the resulting estimators<a href="#fn16" class="footnote-ref" id="fnref16"><sup>16</sup></a>.</p>
<p>Yet, the dimension <span class="math inline">\(D_{\mathit{G}} + D_{\mathit{M}} + D_I\)</span> involved in Problem  to
estimate <span class="math inline">\(\boldsymbol{\gamma}_{\mathit{G}}\)</span>, <span class="math inline">\(\boldsymbol{\gamma}_{\mathit{M}}\)</span> and <span class="math inline">\(\boldsymbol{\Delta}_{\mathit{G}\mathit{M}}\)</span> may be
large especially for applications with an important number of variables
such as in biology with genomic and metagenomic data. To reduce the
dimension, we propose to compress the data according to an underlying
structure which may be defined thanks to a prior knowledge or be
uncovered with clustering algorithms.</p>
</div>
<div id="compressdata" class="section level3">
<h3><span class="header-section-number">5.2.3</span> Compact model</h3>
<p>Assuming we are given a compression function for each group of <span class="math inline">\(\mathit{G}\)</span> and
<span class="math inline">\(\mathit{M}\)</span>, we can shape Problem  into a compact form
<span class="math display">\[\label{eq:compact_detailled_model}  
  y_i = 
  \sum_{g \in \mathcal{G}} \tilde{x}_i^g \beta_{g} +
  \sum_{m \in \mathcal{M}} \tilde{x}_i^m \beta_{m} +
  \sum_{g \in \mathcal{G}} \sum_{m \in \mathcal{M}}
  \underbrace{\left(\tilde{x}_i^g  \cdot
      \tilde{x}_i^m\right)}_{\boldsymbol{\phi}^{gm}_i}
  {\theta_{gm}} + \, \epsilon_i \,,\]</span> where
<span class="math inline">\(\tilde{x}_i^g \in \mathbb{R}\)</span> is the <span class="math inline">\(i^{th}\)</span> compressed sample of the
variables that belong to the group <span class="math inline">\(g\)</span> for the view <span class="math inline">\(\mathit{G}\)</span> and
<span class="math inline">\(\beta_{g} \in \mathbb{R}\)</span> is its corresponding coefficient. The
counterparts on the group <span class="math inline">\(m\)</span> for the view <span class="math inline">\(\mathit{M}\)</span> are
<span class="math inline">\(\tilde{x}_i^m \in \mathbb{R}\)</span> and <span class="math inline">\(\beta_{m} \in \mathbb{R}\)</span>. Finally,
<span class="math inline">\(\theta_{gm} \in \mathbb{R}\)</span> is the interaction between groups <span class="math inline">\(g\)</span> and
<span class="math inline">\(m\)</span>.</p>
<p>We can reformulate Problem  in a vector form. Let
<span class="math inline">\(\tilde{\mathbf{x}}_{i}^{\mathit{G}} \in \mathbb{R}^{N_{\mathit{G}}}\)</span>, <span class="math inline">\(\boldsymbol{\beta}_{\mathit{G}} \in \mathbb{R}^{N_{\mathit{G}}}\)</span>,
<span class="math inline">\(\tilde{\mathbf{x}}_{i}^{\mathit{M}} \in \mathbb{R}^{N_{\mathit{M}}}\)</span> and <span class="math inline">\(\boldsymbol{\beta}_{\mathit{M}} \in \mathbb{R}^{N_{\mathit{M}}}\)</span>
be <span class="math display">\[\begin{aligned}
  \tilde{\mathbf{x}}_{i}^{\mathit{G}} &amp; = (\tilde{x}_i^1 \cdots \tilde{x}_i^g \cdots
                   \tilde{x}_i^{N_{\mathit{G}}})\,,
  &amp;  \boldsymbol{\beta}_{\mathit{G}} &amp; = (\beta_{1} \cdots \beta_{g}
                    \cdots \beta_{N_{\mathit{G}}})^{T}\,,\\
  \tilde{\mathbf{x}}_{i}^{\mathit{M}} &amp; = (\tilde{x}_i^1 \cdots \tilde{x}_i^m \cdots
                   \tilde{x}_i^{N_{\mathit{M}}}) \,,
  &amp; \boldsymbol{\beta}_{\mathit{M}} &amp; = (\beta_{1} \cdots \beta_{m}
                   \cdots \beta_{N_{\mathit{M}}})^{T}\,.\end{aligned}\]</span></p>
<p>We denote by <span class="math inline">\(\boldsymbol{\phi}_{i} \in \mathbb{R}^{N_I}\)</span>, the vector whose general
component is given by <span class="math inline">\(\boldsymbol{\phi}^{gm}_i\)</span> in Equation , that is
<span class="math display">\[\begin{aligned}
  \boldsymbol{\phi}_{i} &amp; =  
                       \left(
                       \boldsymbol{\phi}^{11}_i \cdots 
                       \boldsymbol{\phi}^{1N_{\mathit{M}}}_i
                       \cdots
                       \boldsymbol{\phi}^{g m}_i
                       \cdots 
                       \boldsymbol{\phi}^{{N_{\mathit{G}}} 1}_i
                       \cdots
                       \boldsymbol{\phi}^{{N_{\mathit{G}}}  {N_{\mathit{M}}}}_i
                       \right) \,,\end{aligned}\]</span> and
<span class="math inline">\(\boldsymbol{\theta} \in \mathbb{R}^{N_I}\)</span>, the corresponding vector of coefficients, by
<span class="math display">\[\begin{aligned}
  \boldsymbol{\theta} = &amp; \left(\theta_{11}   \cdots \theta_{1 
              {N_{\mathit{M}}}}  \cdots \theta_{g m}
              \cdots \theta_{{N_{\mathit{G}}} 1}  \cdots
              \theta_{{N_{\mathit{G}}}  {N_{\mathit{M}}}} \right)^T\,. \end{aligned}\]</span></p>
<p>Finally, Problem  reads as a classical linear regression problem
<span class="math display">\[\label{eq:compact_vector_model}
  y_i =  \tilde{\mathbf{x}}_{i}^{\mathit{G}} \boldsymbol{\beta}_{\mathit{G}}  + 
  \tilde{\mathbf{x}}_{i}^{\mathit{M}} \boldsymbol{\beta}_{\mathit{M}} + 
  \boldsymbol{\phi}_{i}  \boldsymbol{\theta}  +
  \epsilon_i \,,\]</span> of dimension <span class="math inline">\(N_{\mathit{G}} + N_{\mathit{M}} + N_I\)</span>.</p>
</div>
<div id="recoverinteractions" class="section level3">
<h3><span class="header-section-number">5.2.4</span> Recovering relevant interactions</h3>
<p>Compared to Problem  and provided that <span class="math inline">\(N_\mathit{G}\)</span> and <span class="math inline">\(N_\mathit{M}\)</span> are
reasonably lower than <span class="math inline">\(D_{\mathit{G}}\)</span> and <span class="math inline">\(D_{\mathit{M}}\)</span>, the dimension of Problem 
decreases drastically so that it might be solved thanks to an
appropriate optimization algorithm coupled with effective computational
facilities. For instance, <span class="citation">(Donoho and Tsaig <a href="#ref-DT:IEEEIT2008">2008</a>)</span> give an overview of <span class="math inline">\(\ell_1\)</span>
regularized algorithms to solve sparse problems like Lasso, which in our
case could take the form:</p>
<p><span class="math display">\[\begin{aligned}
  \underset{\boldsymbol{\beta}_{\mathit{G}}, \boldsymbol{\beta}_{\mathit{M}}, \boldsymbol{\theta}}{\text{argmin}},
  &amp; \quad \sum_{i=1}^n \left(y_i - \tilde{\mathbf{x}}_{i}^{\mathit{G}} \boldsymbol{\beta}_{\mathit{G}} -
    \tilde{\mathbf{x}}_{i}^{\mathit{M}}  \boldsymbol{\beta}_{\mathit{M}}  - \boldsymbol{\phi}_{i}  \boldsymbol{\theta} \right)^2 + \lambda_{\mathit{G}} \sum_{g=1}^{N_{\mathit{G}}}|\beta_{g}| + \lambda_{\mathit{M}} \sum_{m=1}^{N_{\mathit{M}}} |\beta_{m}|  +  \lambda_I \sum_{g,m=1}^{N_I}  |\theta_{gm}|, \end{aligned}\]</span></p>
<p>with <span class="math inline">\(\lambda_\mathit{G}\)</span>, <span class="math inline">\(\lambda_\mathit{M}\)</span> and <span class="math inline">\(\lambda_I\)</span> being the positive
hyperparameters that respectively control the amount of sparsity related
to coefficients <span class="math inline">\(\boldsymbol{\beta}_\mathit{G}\)</span>, <span class="math inline">\(\boldsymbol{\beta}_\mathit{M}\)</span> and <span class="math inline">\(\boldsymbol{\theta}\)</span>. Still, the
dimension may remain large regarding the dimension <span class="math inline">\(N_\mathit{G} + N_\mathit{M} + N_I\)</span>
compared to the number of observations <span class="math inline">\(N\)</span>. Also, note that without
additional constraints, such a formulation would not induce the
dependences hypothesis (SD) and (WD). For that purpose, one could adapt
the works of <span class="citation">(Bien, Taylor, and Tibshirani <a href="#ref-bien2013lasso">2013</a>; Lim and Hastie <a href="#ref-lim2015learning">2015</a>)</span> or <span class="citation">(She, Wang, and Jiang <a href="#ref-she2016group">2016</a>)</span>
mentioned above. We present in the next Section another way to reduce
further the dimension and fulfil the strong dependency hypothesis.</p>
</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-bien2013lasso">
<p>Bien, Jacob, Jonathan Taylor, and Robert Tibshirani. 2013. “A Lasso for Hierarchical Interactions.” <em>Annals of Statistics</em> 41 (3): 1111.</p>
</div>
<div id="ref-lim2015learning">
<p>Lim, Michael, and Trevor Hastie. 2015. “Learning Interactions via Hierarchical Group-Lasso Regularization.” <em>Journal of Computational and Graphical Statistics</em> 24 (3): 627–54.</p>
</div>
<div id="ref-she2016group">
<p>She, Yiyuan, Zhifeng Wang, and He Jiang. 2016. “Group Regularized Estimation Under Structural Hierarchy.” <em>Journal of the American Statistical Association</em> 113 (521): 445–54.</p>
</div>
<div id="ref-jacob2009group">
<p>Jacob, Laurent, Guillaume Obozinski, and Jean-Philippe Vert. 2009. “Group LAsso with Overlap and Graph LAsso.” In <em>Proceedings of the 26th Annual International Conference on Machine Learning</em>, 433–40. Montreal, Quebec, Canada.</p>
</div>
<div id="ref-DT:IEEEIT2008">
<p>Donoho, David L, and Yaakov Tsaig. 2008. “Fast Solution of-Norm Minimization Problems When the Solution May Be Sparse.” <em>IEEE Transactions on Information Theory</em> 54 (11): 4789–4812.</p>
</div>
</div>
<div class="footnotes">
<hr />
<ol start="15">
<li id="fn15"><p>For the sake of clarity, we use these lightened notations which
slightly differ from those used in the previous chapters.<a href="modelsicomore.html#fnref15" class="footnote-back">↩</a></p></li>
<li id="fn16"><p>To our knowledge, their implementation based on alternating
direction method of multipliers is not publicly available.<a href="modelsicomore.html#fnref16" class="footnote-back">↩</a></p></li>
</ol>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="introduction-3.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="implementation.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"download": ["book.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
