<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>References | book.utf8.md</title>
  <meta name="description" content="PdD thesis">
  <meta name="generator" content="bookdown  and GitBook 2.6.7">

  <meta property="og:title" content="References | book.utf8.md />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="PdD thesis" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="References | book.utf8.md />
  
  <meta name="twitter:description" content="PdD thesis" />
  




  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="Bspline.html">

<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />










<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Statistical learning for omics association and interactions studies based on blockwise feature compression</a></li>

<li class="divider"></li>
<li><a href="index.html#section"></a></li>
<li class="chapter" data-level="" data-path="abstract.html"><a href="abstract.html"><i class="fa fa-check"></i>Abstract</a></li>
<li class="chapter" data-level="" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i>General introduction</a></li>
<li class="chapter" data-level="" data-path="notations.html"><a href="notations.html"><i class="fa fa-check"></i>Notations</a></li>
<li class="chapter" data-level="" data-path="abbreviations.html"><a href="abbreviations.html"><i class="fa fa-check"></i>Abbreviations</a></li>
<li class="chapter" data-level="1" data-path="genet.html"><a href="genet.html"><i class="fa fa-check"></i><b>1</b> Basic concepts of molecular genetics</a><ul>
<li class="chapter" data-level="1.1" data-path="genome.html"><a href="genome.html"><i class="fa fa-check"></i><b>1.1</b> Genome description</a></li>
<li class="chapter" data-level="1.2" data-path="genome-sequencing.html"><a href="genome-sequencing.html"><i class="fa fa-check"></i><b>1.2</b> Genome sequencing</a><ul>
<li class="chapter" data-level="1.2.1" data-path="genome-sequencing.html"><a href="genome-sequencing.html#DNAseq"><i class="fa fa-check"></i><b>1.2.1</b> DNA sequencing</a></li>
<li class="chapter" data-level="1.2.2" data-path="genome-sequencing.html"><a href="genome-sequencing.html#sequence-assembly"><i class="fa fa-check"></i><b>1.2.2</b> Sequence assembly</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="dna-polymorphism.html"><a href="dna-polymorphism.html"><i class="fa fa-check"></i><b>1.3</b> DNA polymorphism</a><ul>
<li class="chapter" data-level="1.3.1" data-path="dna-polymorphism.html"><a href="dna-polymorphism.html#restriction-fragment-length-polymorphisms-rflp"><i class="fa fa-check"></i><b>1.3.1</b> Restriction Fragment Length Polymorphisms (RFLP)</a></li>
<li class="chapter" data-level="1.3.2" data-path="dna-polymorphism.html"><a href="dna-polymorphism.html#simple-sequence-length-polymorphisms-sslp"><i class="fa fa-check"></i><b>1.3.2</b> Simple Sequence Length Polymorphisms (SSLP)</a></li>
<li class="chapter" data-level="1.3.3" data-path="dna-polymorphism.html"><a href="dna-polymorphism.html#single-nucleotide-polymorphisms-snp"><i class="fa fa-check"></i><b>1.3.3</b> Single Nucleotide Polymorphisms (SNP)</a></li>
</ul></li>
<li class="chapter" data-level="1.4" data-path="linkage.html"><a href="linkage.html"><i class="fa fa-check"></i><b>1.4</b> Linkage and partial linkage for genetic mapping</a></li>
<li class="chapter" data-level="1.5" data-path="basic-concepts-in-population-genetics.html"><a href="basic-concepts-in-population-genetics.html"><i class="fa fa-check"></i><b>1.5</b> Basic concepts in population genetics</a><ul>
<li class="chapter" data-level="1.5.1" data-path="basic-concepts-in-population-genetics.html"><a href="basic-concepts-in-population-genetics.html#HWE"><i class="fa fa-check"></i><b>1.5.1</b> Hardy-Weinberg equilibrium in large population</a></li>
<li class="chapter" data-level="1.5.2" data-path="basic-concepts-in-population-genetics.html"><a href="basic-concepts-in-population-genetics.html#genetic-drift-in-small-population"><i class="fa fa-check"></i><b>1.5.2</b> Genetic drift in small population</a></li>
<li class="chapter" data-level="1.5.3" data-path="basic-concepts-in-population-genetics.html"><a href="basic-concepts-in-population-genetics.html#concept-of-heritability"><i class="fa fa-check"></i><b>1.5.3</b> Concept of heritability</a></li>
</ul></li>
<li class="chapter" data-level="1.6" data-path="LD.html"><a href="LD.html"><i class="fa fa-check"></i><b>1.6</b> Linkage disequilibrium</a><ul>
<li class="chapter" data-level="1.6.1" data-path="LD.html"><a href="LD.html#definition"><i class="fa fa-check"></i><b>1.6.1</b> Definition</a></li>
<li class="chapter" data-level="1.6.2" data-path="LD.html"><a href="LD.html#measure-of-ld"><i class="fa fa-check"></i><b>1.6.2</b> Measure of LD</a></li>
<li class="chapter" data-level="1.6.3" data-path="LD.html"><a href="LD.html#estimation-of-linkage-disequilibrium"><i class="fa fa-check"></i><b>1.6.3</b> Estimation of linkage disequilibrium</a></li>
<li class="chapter" data-level="1.6.4" data-path="LD.html"><a href="LD.html#origins-of-linkage-disequilibrium"><i class="fa fa-check"></i><b>1.6.4</b> Origins of linkage disequilibrium</a></li>
</ul></li>
<li class="chapter" data-level="1.7" data-path="haplo.html"><a href="haplo.html"><i class="fa fa-check"></i><b>1.7</b> Structure of haplotype blocks in the human genome</a><ul>
<li class="chapter" data-level="" data-path="haplo.html"><a href="haplo.html#definition-of-haplotype-blocks"><i class="fa fa-check"></i>Definition of haplotype blocks</a></li>
<li class="chapter" data-level="" data-path="haplo.html"><a href="haplo.html#patterns-in-human-genome"><i class="fa fa-check"></i>Patterns in human genome</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="stat.html"><a href="stat.html"><i class="fa fa-check"></i><b>2</b> Statistical context</a><ul>
<li class="chapter" data-level="2.1" data-path="notations-1.html"><a href="notations-1.html"><i class="fa fa-check"></i><b>2.1</b> Notations</a></li>
<li class="chapter" data-level="2.2" data-path="concepts-of-statistical-learning.html"><a href="concepts-of-statistical-learning.html"><i class="fa fa-check"></i><b>2.2</b> Concepts of statistical learning</a><ul>
<li class="chapter" data-level="2.2.1" data-path="concepts-of-statistical-learning.html"><a href="concepts-of-statistical-learning.html#prediction"><i class="fa fa-check"></i><b>2.2.1</b> Prediction</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="parametric.html"><a href="parametric.html"><i class="fa fa-check"></i><b>2.3</b> Parametric methods</a><ul>
<li class="chapter" data-level="2.3.1" data-path="parametric.html"><a href="parametric.html#linmod"><i class="fa fa-check"></i><b>2.3.1</b> Linear models</a></li>
<li class="chapter" data-level="2.3.2" data-path="parametric.html"><a href="parametric.html#penalized"><i class="fa fa-check"></i><b>2.3.2</b> Penalized linear regression</a></li>
<li class="chapter" data-level="2.3.3" data-path="parametric.html"><a href="parametric.html#glm"><i class="fa fa-check"></i><b>2.3.3</b> Generalized linear models</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="non-parametric.html"><a href="non-parametric.html"><i class="fa fa-check"></i><b>2.4</b> Splines and generalized additive models: Moving beyond linearit</a><ul>
<li class="chapter" data-level="2.4.1" data-path="non-parametric.html"><a href="non-parametric.html#introduction"><i class="fa fa-check"></i><b>2.4.1</b> Introduction</a></li>
<li class="chapter" data-level="2.4.2" data-path="non-parametric.html"><a href="non-parametric.html#splines"><i class="fa fa-check"></i><b>2.4.2</b> Regression splines</a></li>
<li class="chapter" data-level="2.4.3" data-path="non-parametric.html"><a href="non-parametric.html#mathrmb-splines"><i class="fa fa-check"></i><b>2.4.3</b> <span class="math inline">\(\mathrm{B}\)</span>-splines</a></li>
<li class="chapter" data-level="2.4.4" data-path="non-parametric.html"><a href="non-parametric.html#smoothing"><i class="fa fa-check"></i><b>2.4.4</b> Cubic smoothing splines</a></li>
<li class="chapter" data-level="2.4.5" data-path="non-parametric.html"><a href="non-parametric.html#gam"><i class="fa fa-check"></i><b>2.4.5</b> Generalized additive models (GAM)</a></li>
<li class="chapter" data-level="2.4.6" data-path="non-parametric.html"><a href="non-parametric.html#hgam"><i class="fa fa-check"></i><b>2.4.6</b> High-dimensional generalized additive models (HGAM)</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="combining-cluster-analysis-and-variable-selection.html"><a href="combining-cluster-analysis-and-variable-selection.html"><i class="fa fa-check"></i><b>2.5</b> Combining cluster analysis and variable selection</a><ul>
<li class="chapter" data-level="2.5.1" data-path="combining-cluster-analysis-and-variable-selection.html"><a href="combining-cluster-analysis-and-variable-selection.html#CAH"><i class="fa fa-check"></i><b>2.5.1</b> Hierarchical clustering</a></li>
<li class="chapter" data-level="2.5.2" data-path="combining-cluster-analysis-and-variable-selection.html"><a href="combining-cluster-analysis-and-variable-selection.html#HCAR"><i class="fa fa-check"></i><b>2.5.2</b> Hierarchical Clustering and Averaging Regression</a></li>
<li class="chapter" data-level="2.5.3" data-path="combining-cluster-analysis-and-variable-selection.html"><a href="combining-cluster-analysis-and-variable-selection.html#MLGL"><i class="fa fa-check"></i><b>2.5.3</b> Multi-Layer Group-Lasso (MLGL)</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="hypothesis.html"><a href="hypothesis.html"><i class="fa fa-check"></i><b>2.6</b> Statistical testing of significance</a><ul>
<li class="chapter" data-level="2.6.1" data-path="hypothesis.html"><a href="hypothesis.html#introduction-1"><i class="fa fa-check"></i><b>2.6.1</b> Introduction</a></li>
<li class="chapter" data-level="2.6.2" data-path="hypothesis.html"><a href="hypothesis.html#chi2"><i class="fa fa-check"></i><b>2.6.2</b> <span class="math inline">\(\chi^2\)</span> test</a></li>
<li class="chapter" data-level="2.6.3" data-path="hypothesis.html"><a href="hypothesis.html#LRT"><i class="fa fa-check"></i><b>2.6.3</b> Likelihood ratio test</a></li>
<li class="chapter" data-level="2.6.4" data-path="hypothesis.html"><a href="hypothesis.html#pvalGAM"><i class="fa fa-check"></i><b>2.6.4</b> Calculation of <em>p</em>-values in GAM</a></li>
<li class="chapter" data-level="2.6.5" data-path="hypothesis.html"><a href="hypothesis.html#multiple"><i class="fa fa-check"></i><b>2.6.5</b> Multiple testing comparison</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="asso.html"><a href="asso.html"><i class="fa fa-check"></i><b>3</b> Genome-Wide Association Studies</a><ul>
<li class="chapter" data-level="3.1" data-path="introduction-2.html"><a href="introduction-2.html"><i class="fa fa-check"></i><b>3.1</b> Introduction</a></li>
<li class="chapter" data-level="3.2" data-path="genoquality.html"><a href="genoquality.html"><i class="fa fa-check"></i><b>3.2</b> Genotype quality control</a><ul>
<li class="chapter" data-level="3.2.1" data-path="genoquality.html"><a href="genoquality.html#deviation-from-hwe."><i class="fa fa-check"></i><b>3.2.1</b> Deviation from HWE.</a></li>
<li class="chapter" data-level="3.2.2" data-path="genoquality.html"><a href="genoquality.html#missing-data."><i class="fa fa-check"></i><b>3.2.2</b> Missing data.</a></li>
<li class="chapter" data-level="3.2.3" data-path="genoquality.html"><a href="genoquality.html#distribution-of-test-statistics."><i class="fa fa-check"></i><b>3.2.3</b> Distribution of test statistics.</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="OR.html"><a href="OR.html"><i class="fa fa-check"></i><b>3.3</b> Disease penetrance and odds ratio</a></li>
<li class="chapter" data-level="3.4" data-path="SMA.html"><a href="SMA.html"><i class="fa fa-check"></i><b>3.4</b> Single Marker Analysis</a><ul>
<li class="chapter" data-level="3.4.1" data-path="SMA.html"><a href="SMA.html#pearsons-chi2-statistic"><i class="fa fa-check"></i><b>3.4.1</b> Pearson’s <span class="math inline">\(\chi^2\)</span> statistic</a></li>
<li class="chapter" data-level="3.4.2" data-path="SMA.html"><a href="SMA.html#cochran-armitage-trend-test"><i class="fa fa-check"></i><b>3.4.2</b> Cochran-Armitage trend test</a></li>
<li class="chapter" data-level="3.4.3" data-path="SMA.html"><a href="SMA.html#logitGWAS"><i class="fa fa-check"></i><b>3.4.3</b> Logistic regression and likelihood ratio test</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="GWASlimits.html"><a href="GWASlimits.html"><i class="fa fa-check"></i><b>3.5</b> Limitations</a></li>
<li class="chapter" data-level="3.6" data-path="popstructure.html"><a href="popstructure.html"><i class="fa fa-check"></i><b>3.6</b> Population structure</a><ul>
<li class="chapter" data-level="3.6.1" data-path="popstructure.html"><a href="popstructure.html#genomic-control"><i class="fa fa-check"></i><b>3.6.1</b> Genomic control</a></li>
<li class="chapter" data-level="3.6.2" data-path="popstructure.html"><a href="popstructure.html#structured-association"><i class="fa fa-check"></i><b>3.6.2</b> Structured association</a></li>
<li class="chapter" data-level="3.6.3" data-path="popstructure.html"><a href="popstructure.html#PCC"><i class="fa fa-check"></i><b>3.6.3</b> Principle components correction</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="multiloc.html"><a href="multiloc.html"><i class="fa fa-check"></i><b>3.7</b> Multi-locus analysis</a><ul>
<li class="chapter" data-level="3.7.1" data-path="multiloc.html"><a href="multiloc.html#haplotype-based-approaches"><i class="fa fa-check"></i><b>3.7.1</b> Haplotype-based approaches</a></li>
<li class="chapter" data-level="3.7.2" data-path="multiloc.html"><a href="multiloc.html#rare-variant"><i class="fa fa-check"></i><b>3.7.2</b> Rare-variant association analysis</a></li>
<li class="chapter" data-level="3.7.3" data-path="multiloc.html"><a href="multiloc.html#adjclust"><i class="fa fa-check"></i><b>3.7.3</b> LD based approach to variable selection in GWAS</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="LEOS.html"><a href="LEOS.html"><i class="fa fa-check"></i><b>4</b> Learning the Optimal in GWAS through hierarchical SNP aggregation</a><ul>
<li class="chapter" data-level="4.1" data-path="introleos.html"><a href="introleos.html"><i class="fa fa-check"></i><b>4.1</b> Related work</a></li>
<li class="chapter" data-level="4.2" data-path="leosmethod.html"><a href="leosmethod.html"><i class="fa fa-check"></i><b>4.2</b> Method</a><ul>
<li class="chapter" data-level="4.2.1" data-path="leosmethod.html"><a href="leosmethod.html#CHAC"><i class="fa fa-check"></i><b>4.2.1</b> Step 1. Constrained-HAC</a></li>
<li class="chapter" data-level="4.2.2" data-path="leosmethod.html"><a href="leosmethod.html#Dstar"><i class="fa fa-check"></i><b>4.2.2</b> Step 2. Dimension reduction function</a></li>
<li class="chapter" data-level="4.2.3" data-path="leosmethod.html"><a href="leosmethod.html#cutree"><i class="fa fa-check"></i><b>4.2.3</b> Step 3. Optimal number of groups estimation</a></li>
<li class="chapter" data-level="4.2.4" data-path="leosmethod.html"><a href="leosmethod.html#step4"><i class="fa fa-check"></i><b>4.2.4</b> Step 4. Multiple testing on aggregated-SNP variables</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="numsim.html"><a href="numsim.html"><i class="fa fa-check"></i><b>4.3</b> Numerical simulations</a><ul>
<li class="chapter" data-level="4.3.1" data-path="numsim.html"><a href="numsim.html#simupheno"><i class="fa fa-check"></i><b>4.3.1</b> Simulation of the case-control phenotype</a></li>
<li class="chapter" data-level="4.3.2" data-path="numsim.html"><a href="numsim.html#perfeval"><i class="fa fa-check"></i><b>4.3.2</b> Performance evaluation</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="results.html"><a href="results.html"><i class="fa fa-check"></i><b>4.4</b> Results</a><ul>
<li class="chapter" data-level="4.4.1" data-path="results.html"><a href="results.html#results-and-discussions-of-the-numerical-simulations"><i class="fa fa-check"></i><b>4.4.1</b> Results and discussions of the numerical simulations</a></li>
<li class="chapter" data-level="4.4.2" data-path="results.html"><a href="results.html#performance-results-for-simulated-data."><i class="fa fa-check"></i><b>4.4.2</b> Performance results for simulated data.</a></li>
<li class="chapter" data-level="4.4.3" data-path="results.html"><a href="results.html#leosappli"><i class="fa fa-check"></i><b>4.4.3</b> Application in Wellcome Trust Case Control Consortium(WTCCC) and Ankylosing Spondylitis (AS) studies</a></li>
<li class="chapter" data-level="4.4.4" data-path="results.html"><a href="results.html#realdata"><i class="fa fa-check"></i><b>4.4.4</b> Results in WTCCC and AS studies</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="leosgam.html"><a href="leosgam.html"><i class="fa fa-check"></i><b>4.5</b> Generalized additive models in GWAS</a><ul>
<li class="chapter" data-level="4.5.1" data-path="leosgam.html"><a href="leosgam.html#comparison-of-predictive-power"><i class="fa fa-check"></i><b>4.5.1</b> Comparison of predictive power</a></li>
<li class="chapter" data-level="4.5.2" data-path="leosgam.html"><a href="leosgam.html#results-of-univariate-smoothing-splines-on-aggregated-snp"><i class="fa fa-check"></i><b>4.5.2</b> Results of univariate smoothing splines on aggregated-SNP</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="discussions.html"><a href="discussions.html"><i class="fa fa-check"></i><b>4.6</b> Discussion</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="sicomore.html"><a href="sicomore.html"><i class="fa fa-check"></i><b>5</b> Selection of interaction effects in compressed multiple omics representation</a><ul>
<li class="chapter" data-level="5.1" data-path="introduction-3.html"><a href="introduction-3.html"><i class="fa fa-check"></i><b>5.1</b> Introduction</a><ul>
<li class="chapter" data-level="5.1.1" data-path="introduction-3.html"><a href="introduction-3.html#background-1"><i class="fa fa-check"></i><b>5.1.1</b> Background</a></li>
<li class="chapter" data-level="5.1.2" data-path="introduction-3.html"><a href="introduction-3.html#combining-genome-and-metagenome-analyses."><i class="fa fa-check"></i><b>5.1.2</b> Combining genome and metagenome analyses.</a></li>
<li class="chapter" data-level="5.1.3" data-path="introduction-3.html"><a href="introduction-3.html#taking-structures-into-account-in-association-studies."><i class="fa fa-check"></i><b>5.1.3</b> Taking structures into account in association studies.</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="modelsicomore.html"><a href="modelsicomore.html"><i class="fa fa-check"></i><b>5.2</b> Learning with complementary datasets</a><ul>
<li class="chapter" data-level="5.2.1" data-path="modelsicomore.html"><a href="modelsicomore.html#setting-and-notations"><i class="fa fa-check"></i><b>5.2.1</b> Setting and notations</a></li>
<li class="chapter" data-level="5.2.2" data-path="modelsicomore.html"><a href="modelsicomore.html#interactions-in-linear-models"><i class="fa fa-check"></i><b>5.2.2</b> Interactions in linear models</a></li>
<li class="chapter" data-level="5.2.3" data-path="modelsicomore.html"><a href="modelsicomore.html#compressdata"><i class="fa fa-check"></i><b>5.2.3</b> Compact model</a></li>
<li class="chapter" data-level="5.2.4" data-path="modelsicomore.html"><a href="modelsicomore.html#recoverinteractions"><i class="fa fa-check"></i><b>5.2.4</b> Recovering relevant interactions</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="implementation.html"><a href="implementation.html"><i class="fa fa-check"></i><b>5.3</b> Method</a><ul>
<li class="chapter" data-level="5.3.1" data-path="implementation.html"><a href="implementation.html#preprocess"><i class="fa fa-check"></i><b>5.3.1</b> Preprocessing of the data</a></li>
<li class="chapter" data-level="5.3.2" data-path="implementation.html"><a href="implementation.html#preprocessing-of-metagenomic-data"><i class="fa fa-check"></i><b>5.3.2</b> Preprocessing of metagenomic data</a></li>
<li class="chapter" data-level="5.3.3" data-path="implementation.html"><a href="implementation.html#structure"><i class="fa fa-check"></i><b>5.3.3</b> Structuring the data</a></li>
<li class="chapter" data-level="5.3.4" data-path="implementation.html"><a href="implementation.html#using-the-structure-efficiently"><i class="fa fa-check"></i><b>5.3.4</b> Using the structure efficiently</a></li>
<li class="chapter" data-level="5.3.5" data-path="implementation.html"><a href="implementation.html#identification-of-relevant-supervariables"><i class="fa fa-check"></i><b>5.3.5</b> Identification of relevant supervariables</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="XPsimu.html"><a href="XPsimu.html"><i class="fa fa-check"></i><b>5.4</b> Numerical simulations</a><ul>
<li class="chapter" data-level="5.4.1" data-path="XPsimu.html"><a href="XPsimu.html#data-generation"><i class="fa fa-check"></i><b>5.4.1</b> Data generation</a></li>
<li class="chapter" data-level="5.4.2" data-path="XPsimu.html"><a href="XPsimu.html#generation-of-the-phenotype"><i class="fa fa-check"></i><b>5.4.2</b> Generation of the phenotype</a></li>
<li class="chapter" data-level="5.4.3" data-path="XPsimu.html"><a href="XPsimu.html#comparison-of-methods"><i class="fa fa-check"></i><b>5.4.3</b> Comparison of methods</a></li>
<li class="chapter" data-level="5.4.4" data-path="XPsimu.html"><a href="XPsimu.html#evaluation-metrics"><i class="fa fa-check"></i><b>5.4.4</b> Evaluation metrics</a></li>
<li class="chapter" data-level="5.4.5" data-path="XPsimu.html"><a href="XPsimu.html#performance-results"><i class="fa fa-check"></i><b>5.4.5</b> Performance results</a></li>
<li class="chapter" data-level="5.4.6" data-path="XPsimu.html"><a href="XPsimu.html#computational-time"><i class="fa fa-check"></i><b>5.4.6</b> Computational time</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="XPINRA.html"><a href="XPINRA.html"><i class="fa fa-check"></i><b>5.5</b> Application on real data: rhizosphere of <em>Medicago truncatula</em></a><ul>
<li class="chapter" data-level="5.5.1" data-path="XPINRA.html"><a href="XPINRA.html#material"><i class="fa fa-check"></i><b>5.5.1</b> Material</a></li>
<li class="chapter" data-level="5.5.2" data-path="XPINRA.html"><a href="XPINRA.html#analysis"><i class="fa fa-check"></i><b>5.5.2</b> Analysis</a></li>
<li class="chapter" data-level="5.5.3" data-path="XPINRA.html"><a href="XPINRA.html#results-1"><i class="fa fa-check"></i><b>5.5.3</b> Results</a></li>
<li class="chapter" data-level="5.5.4" data-path="XPINRA.html"><a href="XPINRA.html#results-on-root-shoot-ratio"><i class="fa fa-check"></i><b>5.5.4</b> Results on Root Shoot Ratio</a></li>
<li class="chapter" data-level="5.5.5" data-path="XPINRA.html"><a href="XPINRA.html#results-on-specific-nitrogen-uptake"><i class="fa fa-check"></i><b>5.5.5</b> Results on Specific Nitrogen Uptake</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="discussions-1.html"><a href="discussions-1.html"><i class="fa fa-check"></i><b>5.6</b> Discussions</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="conclusions.html"><a href="conclusions.html"><i class="fa fa-check"></i>Conclusions</a><ul>
<li class="chapter" data-level="" data-path="conclusions.html"><a href="conclusions.html#discussions-on-leos-algorithm"><i class="fa fa-check"></i>Discussions on LEOS algorithm</a></li>
<li class="chapter" data-level="" data-path="conclusions.html"><a href="conclusions.html#discussions-on-sicomore-algorithm"><i class="fa fa-check"></i>Discussions on SICOMORE algorithm</a></li>
<li class="chapter" data-level="" data-path="perspectives.html"><a href="perspectives.html"><i class="fa fa-check"></i>Perspectives</a></li>
</ul></li>
<li class="appendix"><span><b>Annexes</b></span></li>
<li class="chapter" data-level="A" data-path="biasvar.html"><a href="biasvar.html"><i class="fa fa-check"></i><b>A</b> Derivation of the MSE bias-variance decomposition</a></li>
<li class="chapter" data-level="B" data-path="linsmooth.html"><a href="linsmooth.html"><i class="fa fa-check"></i><b>B</b> Linear smoother <span class="citation">(Buja, Hastie, and Tibshirani <span>1989</span>)</span></a></li>
<li class="chapter" data-level="C" data-path="lambdasmooth.html"><a href="lambdasmooth.html"><i class="fa fa-check"></i><b>C</b> Smoothing parameter <span class="math inline">\(\lambda\)</span> for smoothing splines</a></li>
<li class="chapter" data-level="D" data-path="Bspline.html"><a href="Bspline.html"><i class="fa fa-check"></i><b>D</b> <span class="math inline">\(\mathit{B}\)</span>-spline basis</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="references" class="section level1 unnumbered">
<h1>References</h1>

<div id="refs" class="references">
<div>
<p>Aitchison, John, and C. H. Ho. 1989. “The Multivariate Poisson-Log Normal Distribution.” <em>Biometrika</em> 76 (4): 643–53.</p>
</div>
<div>
<p>Aitchison, Jonh. 1982. “The Statistical Analysis of Compositional Data.” <em>Journal of the Royal Statistical Society. Series B (Methodological)</em> 44 (2): 139–77.</p>
</div>
<div>
<p>Aitkin, AC. 1935. “On Least Squares and Linear Combination of Observations.” <em>Proceedings of the Royal Society of Edinburgh</em> 55: 42–48.</p>
</div>
<div>
<p>Anderson, Stephen. 1981. “Shotgun Dna Sequencing Using Cloned Dnase I-Generated Fragments.” <em>Nucleic Acids Research</em> 9 (13): 3015–27.</p>
</div>
<div>
<p>Ardlie, K. G., L. Kruglyak, and M. Seielstad. 2002. “Patterns of Linkage Disequilibrium in the Human Genome.” <em>Nature Reviews Genetics</em> 3 (4): 299–309.</p>
</div>
<div>
<p>Armitage, Peter. 1955. “Tests for Linear Trends in Proportions and Frequencies.” <em>Biometrics</em> 11 (3): 375–86.</p>
</div>
<div>
<p>Asimit, J. L., A. G. Day-Williams, A. P. Morris, and E. Zeggini. 2012. “ARIEL and AMELIA: Testing for an Accumulation of Rare Variants Using Next-Generation Sequencing Data.” <em>Human Heredity</em> 73 (2): 84–94.</p>
</div>
<div>
<p>Bach, Francis R. 2008. “Bolasso: Model Consistent Lasso Estimation Through the Bootstrap.” In <em>Proceedings of the 25th International Conference on Machine Learning</em>, 33–40. ACM.</p>
</div>
<div>
<p>Balding, David J, Martin Bishop, and Chris Cannings. 2008. <em>Handbook of Statistical Genetics</em>. John Wiley &amp; Sons.</p>
</div>
<div>
<p>Barrett, Jeffrey C, and Lon R Cardon. 2006. “Evaluating Coverage of Genome-Wide Association Studies.” <em>Nature Genetics</em> 38 (6): 659.</p>
</div>
<div>
<p>Barrett, Jeffrey C, B Fry, JDMJ Maller, and Mark J Daly. 2004. “Haploview: Analysis and Visualization of Ld and Haplotype Maps.” <em>Bioinformatics</em> 21 (2): 263–65.</p>
</div>
<div>
<p>Benjamini, Yoav, and Yosef Hochberg. 1995. “Controlling the False Discovery Rate: A Practical and Powerful Approach to Multiple Testing.” <em>Journal of the Royal Statistical Society: Series B</em> 57 (1): 289–300.</p>
</div>
<div>
<p>Bien, Jacob, Jonathan Taylor, and Robert Tibshirani. 2013. “A Lasso for Hierarchical Interactions.” <em>Annals of Statistics</em> 41 (3): 1111.</p>
</div>
<div>
<p>Bonferroni, C. 1936. “Teoria Statistica Delle Classi E Calcolo Delle Probabilita.” <em>Pubblicazioni Del R Istituto Superiore Di Scienze Economiche E Commericiali Di Firenze</em> 8: 3–62.</p>
</div>
<div>
<p>Boor, Carl de. 1975. <em>A Practical Guide to Splines</em>. Springer. Springer. <a href="http://www.springer.com/gb/book/9780387953663">http://www.springer.com/gb/book/9780387953663</a>.</p>
</div>
<div>
<p>Bousquet, O., and A. Elisseeff. 2002. “Stability and Generalization.” <em>Journal of Machine Learning Research</em> 2: 499–526.</p>
</div>
<div>
<p>Brown, T.A., D.B.S.T. Brown, T.A. Brown, and L.B.T. Brown. 2007. <em>Genomes 3</em>. Taylor &amp; Francis Group, an Informa Business. Garland Science Pub. <a href="https://books.google.fr/books?id=Cjl98tqp6rsC">https://books.google.fr/books?id=Cjl98tqp6rsC</a>.</p>
</div>
<div>
<p>Brzyski, Damian, Christine B. Peterson, Piotr Sobczyk, Emmanuel J. CandÃšs, Malgorzata Bogdan, and Chiara Sabatti. 2017. “Controlling the Rate of GWAS False Discoveries.” <em>Genetics</em> 205 (January). <a href="https://doi.org/10.1534/genetics.116.193987">https://doi.org/10.1534/genetics.116.193987</a>.</p>
</div>
<div>
<p>Buja, Andreas, Trevor Hastie, and Robert Tibshirani. 1989. “Linear Smoothers and Additive Models.” <em>The Annals of Statistics</em> 17 (2): 453–510. <a href="http://www.jstor.org/stable/2241560">http://www.jstor.org/stable/2241560</a>.</p>
</div>
<div>
<p>Bush, William S, and Jason H Moore. 2012. “Genome-Wide Association Studies.” <em>PLoS Computational Biology</em> 8 (12): e1002822.</p>
</div>
<div>
<p>Caliński, T., and J. Harabasz. 1974. “A Dendrite Method for Cluster Analysis.” <em>Communications in Statistics</em> 3 (1): 1–27.</p>
</div>
<div>
<p>Chapman, Juliet M, Jason D Cooper, John A Todd, and David G Clayton. 2003. “Detecting Disease Associations Due to Linkage Disequilibrium Using Haplotype Tags: A Class of Tests and the Determinants of Statistical Power.” <em>Human Heredity</em> 56 (1-3): 18–31.</p>
</div>
<div>
<p>Charlesworth, Brian, MT Morgan, and D Charlesworth. 1993. “The Effect of Deleterious Mutations on Neutral Molecular Variation.” <em>Genetics</em> 134 (4): 1289–1303.</p>
</div>
<div>
<p>Chen, Wei, Clarence K. Zhang, Yongmei Cheng, Shaowu Zhang, and Hongyu Zhao. 2013. “A Comparison of Methods for Clustering 16S rRNA Sequences into OTUs.” <em>PLoS ONE</em> 8 (8): e70837.</p>
</div>
<div>
<p>Clarke, Geraldine M, Carl A Anderson, Fredrik H Pettersson, Lon R Cardon, Andrew P Morris, and Krina T Zondervan. 2011. “Basic Statistical Analysis in Genetic Case-Control Studies.” <em>Nature Protocols</em> 6 (2): 121.</p>
</div>
<div>
<p>Clayton, David G, Neil M Walker, Deborah J Smyth, Rebecca Pask, Jason D Cooper, Lisa M Maier, Luc J Smink, et al. 2005. “Population Structure, Differential Bias and Genomic Control in a Large-Scale, Case-Control Association Study.” <em>Nature Genetics</em> 37 (11): 1243.</p>
</div>
<div>
<p>Cochran, William G. 1954. “Some Methods for Strengthening the Common <span class="math inline">\(\chi\)</span> 2 Tests.” <em>Biometrics</em> 10 (4): 417–51.</p>
</div>
<div>
<p>Cox, David R. 1958. “The Regression Analysis of Binary Sequences.” <em>Journal of the Royal Statistical Society. Series B (Methodological)</em>, 215–42.</p>
</div>
<div>
<p>Crick, F.H.C., and J.D. Watson. 1954. “The Complementary Structure of Deoxyribonucleic Acid.” <em>Proceedings of the Royal Society of London. Series A. Mathematical and Physical Sciences</em> 223 (1152): 80–96.</p>
</div>
<div>
<p>Daly, Mark J, John D Rioux, Stephen F Schaffner, Thomas J Hudson, and Eric S Lander. 2001. “High-Resolution Haplotype Structure in the Human Genome.” <em>Nature Genetics</em> 29 (2): 229.</p>
</div>
<div>
<p>Dandine-Roulland, Claire, and Herve Perdry. 2015. “The Use of the Linear Mixed Model in Human Genetics.” <em>Human Heredity</em> 80 (4): 196–206.</p>
</div>
<div>
<p>Davey Smith, George, and Shah Ebrahim. 2003. “ÂMendelian Randomizationâ: Can Genetic Epidemiology Contribute to Understanding Environmental Determinants of Disease?” <em>International Journal of Epidemiology</em> 32 (1): 1–22.</p>
</div>
<div>
<p>Davies, Robert B. 1980. “Algorithm as 155: The Distribution of a Linear Combination of <span class="math inline">\(\chi\)</span> 2 Random Variables.” <em>Journal of the Royal Statistical Society. Series C (Applied Statistics)</em> 29 (3): 323–33.</p>
</div>
<div>
<p>Dehman, A., C. Ambroise, and P. Neuvial. 2015. “Performance of a Blockwise Approach in Variable Selection Using Linkage Disequilibrium Information.” <em>BMC Bioinformatics</em> 16: 148.</p>
</div>
<div>
<p>Dehman, Alia. 2015a. “Spatial Clustering of Linkage Disequilibrium Blocks for Genome-Wide Association Studies.” PhD thesis, Université d’Evry Val d’Essonne; Université Paris-Saclay; Laboratoire de Mathématiques et Modélisation d’Evry.</p>
</div>
<div>
<p>———. 2015b. “Spatial Clustering of Linkage Disequilibrium blocks for Genome-Wide Association Studies.” Theses, Université d’Evry Val d’Essonne ; Université Paris-Saclay ; Laboratoire de Mathématiques et Modélisation d’Evry. <a href="https://tel.archives-ouvertes.fr/tel-01288568">https://tel.archives-ouvertes.fr/tel-01288568</a>.</p>
</div>
<div>
<p>Devlin, Bernie, and Kathryn Roeder. 1999. “Genomic Control for Association Studies.” <em>Biometrics</em> 55 (4): 997–1004.</p>
</div>
<div>
<p>Devlin, B, and Neil Risch. 1995. “A Comparison of Linkage Disequilibrium Measures for Fine-Scale Mapping.” <em>Genomics</em> 29 (2): 311–22.</p>
</div>
<div>
<p>Diaz-Quijano, Fredi A. 2012. “A Simple Method for Estimating Relative Risk Using Logistic Regression".” <em>BMC Medical Research Methodology</em> 1 (1): 14.</p>
</div>
<div>
<p>Donoho, David L, and Yaakov Tsaig. 2008. “Fast Solution of-Norm Minimization Problems When the Solution May Be Sparse.” <em>IEEE Transactions on Information Theory</em> 54 (11): 4789–4812.</p>
</div>
<div>
<p>Dudbridge, Frank, and Arief Gusnanto. 2008. “Estimation of Significance Thresholds for Genomewide Association Scans.” <em>Genetic Epidemiology</em> 32 (3): 227–34.</p>
</div>
<div>
<p>Efron, Bradley, Trevor Hastie, Iain Johnstone, Robert Tibshirani, and others. 2004. “Least Angle Regression.” <em>The Annals of Statistics</em> 32 (2): 407–99.</p>
</div>
<div>
<p>Eubank, Randall L. 1999. <em>Nonparametric Regression and Spline Smoothing, Second Edition</em>. CRC Press.</p>
</div>
<div>
<p>Excoffier, Laurent, and Montgomery Slatkin. 1995. “Maximum-Likelihood Estimation of Molecular Haplotype Frequencies in a Diploid Population.” <em>Molecular Biology and Evolution</em> 12 (5): 921–27.</p>
</div>
<div>
<p>Fergus, Paul, Casimiro Curbelo Montanez, Basma Abdulaimma, Paulo Lisboa, and Carl Chalmers. 2018. “Utilising Deep Learning and Genome Wide Association Studies for Epistatic-Driven Preterm Birth Classification in African-American Women.” <em>arXiv Preprint arXiv:1801.02977</em>.</p>
</div>
<div>
<p>Finucane, Hilary K, Brendan Bulik-Sullivan, Alexander Gusev, Gosia Trynka, Yakir Reshef, Po-Ru Loh, Verneri Anttila, et al. 2015. “Partitioning Heritability by Functional Annotation Using Genome-Wide Association Summary Statistics.” <em>Nature Genetics</em> 47 (11): 1228.</p>
</div>
<div>
<p>Fisher, Ronald A. 1919. “XV.âThe Correlation Between Relatives on the Supposition of Mendelian Inheritance.” <em>Earth and Environmental Science Transactions of the Royal Society of Edinburgh</em> 52 (2): 399–433.</p>
</div>
<div>
<p>———. 1935. “Statistical Methods for Research Workers.” <em>Edinburgh: Oliver and Boyd, 1934 and the Logic of Inductive Interence; Royal Statistical Society</em> 98: S–39.</p>
</div>
<div>
<p>Fletcher, Roger. 1987. “Practical Methods of Optimization John Wiley &amp; Sons.” <em>New York</em> 80.</p>
</div>
<div>
<p>Friedman, Jerome, Trevor Hastie, and Robert Tibshirani. 2001. <em>The Elements of Statistical Learning</em>. Vol. 1. Springer series in statistics New York.</p>
</div>
<div>
<p>Frisse, L, RR Hudson, A Bartoszewicz, JD Wall, J Donfack, and A Di Rienzo. 2001. “Gene Conversion and Different Population Histories May Explain the Contrast Between Polymorphism and Linkage Disequilibrium Levels.” <em>The American Journal of Human Genetics</em> 69 (4): 831–43.</p>
</div>
<div>
<p>Gabriel, S. B., S. F. Schaffner, H. Nguyen, J. M. Moore, J. Roy, B. Blumenstiel, J. Higgins, et al. 2002. “The Structure of Haplotype Blocks in the Human Genome.” <em>Science</em> 296 (5576): 2225–9.</p>
</div>
<div>
<p>Garibyan, Lilit, and Nidhi Avashia. 2013. “Research Techniques Made Simple: Polymerase Chain Reaction (Pcr).” <em>The Journal of Investigative Dermatology</em> 133 (3): e6.</p>
</div>
<div>
<p>Gauss, Carl Friedrich. 1809. <em>Theoria Motus Corporum Coelestium in Sectionibus Conicis Solem Ambientium Auctore Carolo Friderico Gauss</em>. sumtibus Frid. Perthes et IH Besser.</p>
</div>
<div>
<p>Geurst, P., V. Botta, and G. Louppe. 2014. “Exploiting SNP Correlations Within Random Forest for Genome-Wide Association Studies.” <em>PLoS ONE</em> 9 (4).</p>
</div>
<div>
<p>Goeman, Jelle J, Aldo Solari, and others. 2011. “Multiple Testing for Exploratory Research.” <em>Statistical Science</em> 26 (4): 584–97.</p>
</div>
<div>
<p>Golub, Gene H, Michael Heath, and Grace Wahba. 1979. “Generalized Cross-Validation as a Method for Choosing a Good Ridge Parameter.” <em>Technometrics</em> 21 (2): 215–23.</p>
</div>
<div>
<p>Gordon, Aharon D. 1999. <em>Classification</em>. Monographs on Statistics and Applied Probability. Chapman &amp; Hall.</p>
</div>
<div>
<p>Green, Peter J., and Brian S. Yandell. 1985. <em>Semi-Parametric Generalized Linear Models</em>. Lecture Notes in Statistics. Springer, New York, NY. <a href="https://link.springer.com/chapter/10.1007/978-1-4615-7070-7_6">https://link.springer.com/chapter/10.1007/978-1-4615-7070-7_6</a>.</p>
</div>
<div>
<p>Grimm, Eric C. 1987. “CONISS: A Fortran 77 Program for Stratigraphically Constrained Cluster Analysis by the Method of Incremental Sum of Squares.” <em>Computers &amp; Geosciences</em> 13 (1): 13–35.</p>
</div>
<div>
<p>Grimonprez, Quentin. 2016. “Selection de Groupes de Variables corrÃÂ©lÃÂ©es En Grande Dimension.” PhD thesis, Université de Lille; Lille 1.</p>
</div>
<div>
<p>Gu, Chong. 2002. <em>Smoothing Spline ANOVA Models</em>. Springer Series in Statistics. New York, NY: Springer New York. <a href="http://link.springer.com/10.1007/978-1-4757-3683-0">http://link.springer.com/10.1007/978-1-4757-3683-0</a>.</p>
</div>
<div>
<p>Hastie, T. J., and R. J. Tibshirani. 1990. <em>Generalized Additive Models</em>. CRC Press.</p>
</div>
<div>
<p>Hedrick, Philip W. 1987. “Gametic Disequilibrium Measures: Proceed with Caution.” <em>Genetics</em> 117 (2): 331–41.</p>
</div>
<div>
<p>Hill, WG, and Alan Robertson. 1968. “Linkage Disequilibrium in Finite Populations.” <em>Theoretical and Applied Genetics</em> 38 (6): 226–31.</p>
</div>
<div>
<p>Hill, William G. 1974. “Estimation of Linkage Disequilibrium in Randomly Mating Populations.” <em>Heredity</em> 33 (2): 229.</p>
</div>
<div>
<p>Hoerl, Arthur E, and Robert W Kennard. 1970. “Ridge Regression: Biased Estimation for Nonorthogonal Problems.” <em>Technometrics</em> 12 (1): 55–67.</p>
</div>
<div>
<p>Holm, Sture. 1979a. “A Simple Sequentially Rejective Multiple Test Procedure.” <em>Scandinavian Journal of Statistics</em>, 65–70.</p>
</div>
<div>
<p>———. 1979b. “A Simple Sequentially Rejective Multiple Test Procedure.” <em>Scandinavian Journal of Statistics</em> 6 (2): 65–70.</p>
</div>
<div>
<p>Huber, Wolfgang, Anja von Heydebreck, Holger Sültmann, Annemarie Poustka, and Martin Vingron. 2003. “Parameter Estimation for the Calibration and Variance Stabilization of Microarray Data.” <em>Statistical Applications in Genetics and Molecular Biology</em> 2 (1).</p>
</div>
<div>
<p>Hunter, David J. 2005. “Gene–Environment Interactions in Human Diseases.” <em>Nature Reviews Genetics</em> 6 (4): 287.</p>
</div>
<div>
<p>International Genetics of Ankylosing Spondylitis Consortium (IGAS), Adrian Cortes, Johanna Hadler, Jenny P. Pointon, Philip C. Robinson, and others. 2013. “Identification of Multiple Risk Variants for Ankylosing Spondylitis Through High-Density Genotyping of Immune-Related Loci.” <em>Nature Genetics</em> 45 (7): 730–38.</p>
</div>
<div>
<p>Jacob, Laurent, Guillaume Obozinski, and Jean-Philippe Vert. 2009. “Group LAsso with Overlap and Graph LAsso.” In <em>Proceedings of the 26th Annual International Conference on Machine Learning</em>, 433–40. Montreal, Quebec, Canada.</p>
</div>
<div>
<p>Jenatton, Rodolphe, Jean-Yves Audibert, and Francis Bach. 2011. “Structured Variable Selection with Sparsity-Inducing Norms.” <em>Journal of Machine Learning Research</em> 12 (Oct): 2777–2824.</p>
</div>
<div>
<p>Jonsson, Viktor, Tobias Österlund, Olle Nerman, and Erik Kristiansson. 2017. “Variability in Metagenomic Count Data and Its Influence on the Identification of Differentially Abundant Genes.” <em>Journal of Computational Biology</em> 24 (4): 311–26.</p>
</div>
<div>
<p>Kaessmann, Henrik, Sebastian Zöllner, Anna C Gustafsson, Victor Wiebe, Maris Laan, Joakim Lundeberg, Mathias Uhlén, and Svante Pääbo. 2002. “Extensive Linkage Disequilibrium in Small Human Populations in Eurasia.” <em>The American Journal of Human Genetics</em> 70 (3): 673–85.</p>
</div>
<div>
<p>Kang, Hyun Min, Jae Hoon Sul, Susan K Service, Noah A Zaitlen, Sit-yee Kong, Nelson B Freimer, Chiara Sabatti, Eleazar Eskin, and others. 2010. “Variance Component Model to Account for Sample Structure in Genome-Wide Association Studies.” <em>Nature Genetics</em> 42 (4): 348.</p>
</div>
<div>
<p>Klein, Robert J, Caroline Zeiss, Emily Y Chew, Jen-Yue Tsai, Richard S Sackler, Chad Haynes, Alice K Henning, et al. 2005. “Complement Factor H Polymorphism in Age-Related Macular Degeneration.” <em>Science</em> 308 (5720): 385–89.</p>
</div>
<div>
<p>Krzanowski, W. J., and Y. T. Lai. 1988. “A Criterion for Determining the Number of Groups in a Data Set Using Sum-of-Squares Clustering.” <em>Biometrics</em> 44 (1): 23–34.</p>
</div>
<div>
<p>Kwak, Il-Youp, and Wei Pan. 2016. “Adaptive Gene- and Pathway-Trait Association Testing with GWAS Summary Statistics.” <em>Bioinformatics</em> 32: 1178–84. <a href="https://doi.org/10.1093/bioinformatics/btv719">https://doi.org/10.1093/bioinformatics/btv719</a>.</p>
</div>
<div>
<p>Laan, Maris, and Svante Pääbo. 1997. “Demographic History and Linkage Disequilibrium in Human Populations.” <em>Nature Genetics</em> 17 (4): 435.</p>
</div>
<div>
<p>Lander, Eric S. 1996. “The New Genomics: Global Views of Biology.” <em>Science</em> 274 (5287): 536–39.</p>
</div>
<div>
<p>Law, Charity W, Yunshun Chen, Wei Shi, and Gordon K Smyth. 2014. “Voom: Precision Weights Unlock Linear Model Analysis Tools for Rna-Seq Read Counts.” <em>Genome Biology</em> 15 (2): R29.</p>
</div>
<div>
<p>Lee, S., G. R. Abecasis, M. Boehnke, and X. Lin. 2014. “Rare-Variant Association Analysis: Study Designs and Statistical Tests.” <em>American Journal of Human Genetics</em> 95 (1): 5–23.</p>
</div>
<div>
<p>Lee, Seunghak, Nico Görnitz, Eric P. Xing, David Heckerman, and Christoph Lippert. 2017. “Ensembles of LAsso Screening Rules.” <em>IEEE Transactions on Pattern Analysis and Machine Intelligence</em> PP (99): 1–1.</p>
</div>
<div>
<p>Lee, S., M. C. Wu, and X. Lin. 2012. “Optimal Tests for Rare Variant Effects in Sequencing Association Studies.” <em>Biostatistics</em> 13 (4): 762–75.</p>
</div>
<div>
<p>Lewontin, R. C. 1964. “THE Interaction of Selection and Linkage. I. GENERAL Considerations; Heterotic Models.” <em>Genetics</em> 49 (1): 49–67.</p>
</div>
<div>
<p>Li, Bingshan, and Suzanne M Leal. 2008. “Methods for Detecting Associations with Rare Variants for Common Diseases: Application to Analysis of Sequence Data.” <em>The American Journal of Human Genetics</em> 83 (3): 311–21.</p>
</div>
<div>
<p>Lim, Michael, and Trevor Hastie. 2015. “Learning Interactions via Hierarchical Group-Lasso Regularization.” <em>Journal of Computational and Graphical Statistics</em> 24 (3): 627–54.</p>
</div>
<div>
<p>Lin, Xihong. 1997. “Variance Component Testing in Generalised Linear Models with Random Effects.” <em>Biometrika</em> 84 (2): 309–26.</p>
</div>
<div>
<p>Lin, Xinyi, Seunggeun Lee, David C. Christiani, and Xihong Lin. 2013. “Test for Interactions Between a Genetic Marker Set and Environment in Generalized Linear Models.” <em>Biostatistics</em> 14 (4): 667–81.</p>
</div>
<div>
<p>Listgarten, Jennifer, Christoph Lippert, Carl M. Kadie, Robert I. Davidson, Eleazar Eskin, and David Heckerman. 2012. “Improved Linear Mixed Models for Genome-Wide Association Studies.” <em>Nature Methods</em> 9 (6): 525–26. <a href="https://doi.org/10.1038/nmeth.2037">https://doi.org/10.1038/nmeth.2037</a>.</p>
</div>
<div>
<p>Listgarten, J., C. Lippert, E. Y. Kang, J. Xiang, C. M. Kadie, and D. Heckerman. 2013. “A Powerful and Efficient Set Test for Genetic Markers That Handles Confounders.” <em>Bioinformatics</em> 29 (12): 1526–33.</p>
</div>
<div>
<p>Longford, Nicholas T. 1987. “A Fast Scoring Algorithm for Maximum Likelihood Estimation in Unbalanced Mixed Models with Nested Random Effects.” <em>Biometrika</em> 74 (4): 817–27.</p>
</div>
<div>
<p>Love, Michael I, Wolfgang Huber, and Simon Anders. 2014. “Moderated Estimation of Fold Change and Dispersion for Rna-Seq Data with Deseq2.” <em>Genome Biology</em> 15 (12): 550.</p>
</div>
<div>
<p>Maciukiewicz, Malgorzata, Victoria S Marshe, Anne-Christin Hauschild, Jane A Foster, Susan Rotzinger, James L Kennedy, Sidney H Kennedy, Daniel J Müller, and Joseph Geraci. 2018. “GWAS-Based Machine Learning Approach to Predict Duloxetine Response in Major Depressive Disorder.” <em>Journal of Psychiatric Research</em> 99: 62–68.</p>
</div>
<div>
<p>Madsen, Bo Eskerod, and Sharon R Browning. 2009. “A Groupwise Association Test for Rare Mutations Using a Weighted Sum Statistic.” <em>PLoS Genetics</em> 5 (2): e1000384.</p>
</div>
<div>
<p>Maher, B. 2008. “Personal Genomes: The Case of the Missing Heritability.” <em>Nature News</em> 456 (7218): 18–21.</p>
</div>
<div>
<p>Manolio, T. A., and P. M. Visscher. 2009. “Finding the Missing Heritability of Complex Diseases.” <em>Nature</em> 461 (7265): 747–53.</p>
</div>
<div>
<p>Manolio, Teri A, Lisa D Brooks, and Francis S Collins. 2008. “A Hapmap Harvest of Insights into the Genetics of Common Disease.” <em>The Journal of Clinical Investigation</em> 118 (5): 1590–1605.</p>
</div>
<div>
<p>Mary-Huard, Tristan, and Stephane Robin. 2009. “Tailored Aggregation for Classification.” <em>IEEE Transactions on Pattern Analysis and Machine Intelligence</em> 31 (11): 2098–2105.</p>
</div>
<div>
<p>Maxam, Allan M, and Walter Gilbert. 1977. “A New Method for Sequencing Dna.” <em>Proceedings of the National Academy of Sciences</em> 74 (2): 560–64.</p>
</div>
<div>
<p>Mc Cullagh, Peter, and J. A. Nelder. 1989. “Generalized Linear Models, Second Edition.” <em>CRC Press</em>. <a href="https://www.crcpress.com/Generalized-Linear-Models-Second-Edition/McCullagh-Nelder/p/book/9780412317606">https://www.crcpress.com/Generalized-Linear-Models-Second-Edition/McCullagh-Nelder/p/book/9780412317606</a>.</p>
</div>
<div>
<p>Meier, Lukas, Sara van de Geer, and Peter Buhlmann. 2009. “High-Dimensional Additive Modeling.” <em>The Annals of Statistics</em> 37: 3779–3821. <a href="https://doi.org/10.1214/09-AOS692">https://doi.org/10.1214/09-AOS692</a>.</p>
</div>
<div>
<p>Meier, Lukas, Sara Van De Geer, and Peter Buhlmann. 2008. “The Group Lasso for Logistic Regression.” <em>Journal of the Royal Statistical Society: Series B (Statistical Methodology)</em> 70 (1): 53–71. <a href="http://onlinelibrary.wiley.com/doi/10.1111/j.1467-9868.2007.00627.x/abstract">http://onlinelibrary.wiley.com/doi/10.1111/j.1467-9868.2007.00627.x/abstract</a>.</p>
</div>
<div>
<p>Meinshausen, N. 2008. “Hierarchical Testing of Variable Importance.” <em>Biometrika</em> 95 (2): 265–78.</p>
</div>
<div>
<p>Meinshausen, Nicolai, and Peter Bühlmann. 2010. “Stability Selection.” <em>Journal of the Royal Statistical Society: Series B (Statistical Methodology)</em> 72 (4): 417–73.</p>
</div>
<div>
<p>Mendel, Gregor Johann. 1865. “Versuche ãœber Pflanzen-Hybridenâ [Experiments Concerning Plant Hybrids].” <em>Proceedings of the Natural History Society of BrÃŒnn</em> IV: 3–47.</p>
</div>
<div>
<p>Mieth, Bettina, Marius Kloft, Juan Antonio Rodrı́guez, Sören Sonnenburg, Robin Vobruba, Carlos Morcillo-Suárez, Xavier Farré, et al. 2016. “Combining Multiple Hypothesis Testing with Machine Learning Increases the Statistical Power of Genome-Wide Association Studies.” <em>Scientific Reports</em> 6: 36671.</p>
</div>
<div>
<p>Milligan, Glenn W., and Martha C. Cooper. 1985. “An Examination of Procedures for Determining the Number of Clusters in a Data Set.” <em>Psychometrika</em> 50 (2): 159–79.</p>
</div>
<div>
<p>Morgenthaler, Stephan, and William G Thilly. 2007. “A Strategy to Discover Genes That Carry Multi-Allelic or Mono-Allelic Risk for Common Diseases: A Cohort Allelic Sums Test (Cast).” <em>Mutation Research/Fundamental and Molecular Mechanisms of Mutagenesis</em> 615 (1): 28–56.</p>
</div>
<div>
<p>Mukerji, Krisha Gopal, C Manoharachary, and BP Chamola. 2002. <em>Techniques in Mycorrhizal Studies</em>. Springer Science &amp; Business Media.</p>
</div>
<div>
<p>Nelder, J. A., and R. W. M. Wedderburn. 1972. “Generalized Linear Models.” <em>Journal of the Royal Statistical Society: Series A</em> 135 (3): 370–84.</p>
</div>
<div>
<p>Neyman, Jerzy, and Egon S Pearson. 1933. “The Testing of Statistical Hypotheses in Relation to Probabilities a Priori.” In <em>Mathematical Proceedings of the Cambridge Philosophical Society</em>, 29:492–510. Cambridge University Press.</p>
</div>
<div>
<p>Nyrén, Pettersson, Bertil Pettersson, and Mathias Uhlén. 1993. “Solid Phase Dna Minisequencing by an Enzymatic Luminometric Inorganic Pyrophosphate Detection Assay.” <em>Analytical Biochemistry</em> 208 (1): 171–75.</p>
</div>
<div>
<p>Osborne, Michael R, Brett Presnell, and Berwin A Turlach. 2000. “A New Approach to Variable Selection in Least Squares Problems.” <em>IMA Journal of Numerical Analysis</em> 20 (3): 389–403.</p>
</div>
<div>
<p>ParÃ©, Guillaume, Senay Asma, and Wei Q. Deng. 2015. “Contribution of Large Region Joint Associations to Complex Traits Genetics.” <em>PLOS Genetics</em> 11. <a href="https://doi.org/10.1371/journal.pgen.1005103">https://doi.org/10.1371/journal.pgen.1005103</a>.</p>
</div>
<div>
<p>Parikh, Neal, Stephen Boyd, and others. 2014. “Proximal Algorithms.” <em>Foundations and Trends in Optimization</em> 1 (3): 127–239.</p>
</div>
<div>
<p>Park, Mee Young, Trevor Hastie, and Robert Tibshirani. 2007. “Averaged Gene Expressions for Regression.” <em>Biostatistics</em> 8 (2): 212–27.</p>
</div>
<div>
<p>Pearson, Karl. 1896. “Mathematical Contributions to the Theory of Evolution. On a Form of Spurious Correlation Which May Arise When Indices Are Used in the Measurement of Organs.” <em>Proceedings of the Royal Society of London</em> 60: 489–98.</p>
</div>
<div>
<p>———. 1900. “On the Criterion That a Given System of Deviations from the Probable in the Case of a Correlated System of Variables Is Such That It Can Be Reasonably Supposed to Have Arisen from Random Sampling.” <em>The London, Edinburgh, and Dublin Philosophical Magazine and Journal of Science</em> 50 (302): 157–75.</p>
</div>
<div>
<p>Pereira, Mariana Buongermino, Mikael Wallroth, Viktor Jonsson, and Erik Kristiansson. 2018. “Comparison of Normalization Methods for the Analysis of Metagenomic Gene Abundance Data.” <em>BMC Genomics</em> 19 (1): 274.</p>
</div>
<div>
<p>Petersen, Ashley, Carolina Alvarez, Scott DeClaire, and Nathan L. Tintle. 2013. “Assessing Methods for Assigning SNPs to Genes in Gene-Based Tests of Association Using Common Variants.” <em>PLOS ONE</em> 8. <a href="https://doi.org/10.1371/journal.pone.0062161">https://doi.org/10.1371/journal.pone.0062161</a>.</p>
</div>
<div>
<p>Pinton, Roberto, Zeno Varanini, and Paolo Nannipieri. 2007. <em>The Rhizosphere: Biochemistry and Organic Substances at the Soil-Plant Interface</em>. CRC press.</p>
</div>
<div>
<p>Price, A. L., N. J. Patterson, R. M. Plenge, M. E. Weinblatt, N. A. Shadick, and D. Reich. 2006. “Principal Components Analysis Corrects for Stratification in Genome-Wide Association Studies.” <em>Nature Genetics</em> 38: 904–9.</p>
</div>
<div>
<p>Pritchard, Jonathan K, and Molly Przeworski. 2001. “Linkage Disequilibrium in Humans: Models and Data.” <em>The American Journal of Human Genetics</em> 69 (1): 1–14.</p>
</div>
<div>
<p>Pritchard, Jonathan K, Matthew Stephens, and Peter Donnelly. 2000. “Inference of Population Structure Using Multilocus Genotype Data.” <em>Genetics</em> 155 (2): 945–59.</p>
</div>
<div>
<p>Purcell, S., B. Neale, K. Todd-Brown, L. Thomas, M. A. R. Ferreira, D. Bender, J. Maller, et al. 2007. “PLINK: A Tool Set for Whole-Genome Association and Population-Based Linkage Analyses.” <em>The American Journal of Human Genetics</em> 81 (3): 559–75.</p>
</div>
<div>
<p>Qin, Junjie, Yingrui Li, Zhiming Cai, Shenghui Li, Jianfeng Zhu, Fan Zhang, Suisha Liang, et al. 2012. “A Metagenome-Wide Association Study of Gut Microbiota in Type 2 Diabetes.” <em>Nature</em> 490 (7418): 55–60.</p>
</div>
<div>
<p>RA Fisher, MA. 1922. “On the Mathematical Foundations of Theoretical Statistics.” <em>Phil. Trans. R. Soc. Lond. A</em> 222 (594-604): 309–68.</p>
</div>
<div>
<p>Rau, Andrea, Florence Jaffrézic, and Grégory Nuel. 2013. “Joint Estimation of Causal Effects from Observational and Intervention Gene Expression Data.” <em>BMC Systems Biology</em> 7 (1): 111.</p>
</div>
<div>
<p>Rau, Andrea, and Cathy Maugis-Rabusseau. 2017. “Transformation and Model Choice for Rna-Seq Co-Expression Analysis.” <em>Briefings in Bioinformatics</em> 19 (3): 425–36.</p>
</div>
<div>
<p>Reich, David E, Michele Cargill, Stacey Bolk, James Ireland, Pardis C Sabeti, Daniel J Richter, Thomas Lavery, et al. 2001. “Linkage Disequilibrium in the Human Genome.” <em>Nature</em> 411 (6834): 199.</p>
</div>
<div>
<p>Reich, David, Alkes L Price, and Nick Patterson. 2008. “Principal Component Analysis of Genetic Data.” <em>Nature Genetics</em> 40 (5): 491.</p>
</div>
<div>
<p>Reinsch, Christian H. 1967. “Smoothing by Spline Functions.” <em>Numer. Math.</em> 10 (3): 177–83. <a href="http://dx.doi.org/10.1007/BF02162161">http://dx.doi.org/10.1007/BF02162161</a>.</p>
</div>
<div>
<p>Rice, John A. 2006. <em>Mathematical Statistics and Data Analysis</em>. Cengage Learning.</p>
</div>
<div>
<p>Risch, Neil, and Kathleen Merikangas. 1996. “The Future of Genetic Studies of Complex Human Diseases.” <em>Science</em> 273 (5281): 1516–7.</p>
</div>
<div>
<p>Robbins, RB. 1918. “Some Applications of Mathematics to Breeding Problems Iii.” <em>Genetics</em> 3 (4). <a href="http://europepmc.org/articles/PMC1200443">http://europepmc.org/articles/PMC1200443</a>.</p>
</div>
<div>
<p>Robinson, Mark D, and Alicia Oshlack. 2010. “A Scaling Normalization Method for Differential Expression Analysis of Rna-Seq Data.” <em>Genome Biology</em> 11 (3): R25.</p>
</div>
<div>
<p>Rokach, Lior, and Oded Maimon. 2005. “Clustering Methods.” In <em>Data Mining and Knowledge Discovery Handbook</em>, 321–52. Springer.</p>
</div>
<div>
<p>Roquain, Etienne. 2010. “Type I Error Rate Control for Testing Many Hypotheses: A Survey with Proofs.” <em>arXiv Preprint arXiv:1012.4078</em>.</p>
</div>
<div>
<p>Sanger, Frederick, Steven Nicklen, and Alan R Coulson. 1977. “DNA Sequencing with Chain-Terminating Inhibitors.” <em>Proceedings of the National Academy of Sciences</em> 74 (12): 5463–7.</p>
</div>
<div>
<p>Segata, Nicola, Jacques Izard, Levi Waldron, Dirk Gevers, Larisa Miropolsky, Wendy S Garrett, and Curtis Huttenhower. 2011. “Metagenomic Biomarker Discovery and Explanation.” <em>Genome Biology</em> 12 (6): R60.</p>
</div>
<div>
<p>Segura, Vincent, Bjarni J Vilhjálmsson, Alexander Platt, Arthur Korte, Ümit Seren, Quan Long, and Magnus Nordborg. 2012. “An Efficient Multi-Locus Mixed-Model Approach for Genome-Wide Association Studies in Structured Populations.” <em>Nature Genetics</em> 44 (7): 825.</p>
</div>
<div>
<p>Sharpton, Thomas J. 2014. “An Introduction to the Analysis of Shotgun Metagenomic Data.” <em>Frontiers in Plant Science</em> 5: 209.</p>
</div>
<div>
<p>She, Yiyuan, Zhifeng Wang, and He Jiang. 2016. “Group Regularized Estimation Under Structural Hierarchy.” <em>Journal of the American Statistical Association</em> 113 (521): 445–54.</p>
</div>
<div>
<p>Simpson, Edward H. 1951. “The Interpretation of Interaction in Contingency Tables.” <em>Journal of the Royal Statistical Society. Series B (Methodological)</em>, 238–41.</p>
</div>
<div>
<p>Smith, John Maynard, and John Haigh. 1974. “The Hitch-Hiking Effect of a Favourable Gene.” <em>Genetics Research</em> 23 (1): 23–35.</p>
</div>
<div>
<p>Srinivas, Girish, Steffen Möller, Jun Wang, Sven Künzel, Detlef Zillikens, John F Baines, and Saleh M Ibrahim. 2013. “Genome-Wide Mapping of Gene–Microbiota Interactions in Susceptibility to Autoimmune Skin Blistering.” <em>Nature Communications</em> 4.</p>
</div>
<div>
<p>Stanislas, Virginie, Cyril Dalmasso, and Christophe Ambroise. 2017. “Eigen-Epistasis for Detecting Gene-Gene Interactions.” <em>BMC Bioinformatics</em> 18 (1): 54.</p>
</div>
<div>
<p>Stephens, Matthew, Nicholas J Smith, and Peter Donnelly. 2001. “A New Statistical Method for Haplotype Reconstruction from Population Data.” <em>The American Journal of Human Genetics</em> 68 (4): 978–89.</p>
</div>
<div>
<p>Stroup, Walter W. 2012. <em>Generalized Linear Mixed Models: Modern Concepts, Methods and Applications</em>. CRC press.</p>
</div>
<div>
<p>Sturtevant, A.H. 2001. <em>A History of Genetics</em>. G - Reference,Information and Interdisciplinary Subjects Series. Cold Spring Harbor Laboratory Press. <a href="https://books.google.fr/books?id=wDIisw1ZqAMC">https://books.google.fr/books?id=wDIisw1ZqAMC</a>.</p>
</div>
<div>
<p>Su, Z., J. Marchini, and P. Donnelly. 2011. “HAPGEN2: Simulation of Multiple Disease Snps.” <em>Bioinformatics</em> 27 (16): 2304.</p>
</div>
<div>
<p>Šidák, Zbyněk. 1967. “Rectangular Confidence Regions for the Means of Multivariate Normal Distributions.” <em>Journal of the American Statistical Association</em> 62 (318): 626–33.</p>
</div>
<div>
<p>Thomas, Duncan C. 2004. <em>Statistical Methods in Genetic Epidemiology</em>. Oxford University Press.</p>
</div>
<div>
<p>Tibshirani, Robert. 1988. “Estimating Transformations for Regression via Additivity and Variance Stabilization.” <em>Journal of the American Statistical Association</em> 83 (402): 394–405.</p>
</div>
<div>
<p>———. 1996. “Regression Shrinkage and Selection via the Lasso.” <em>Journal of the Royal Statistical Society. Series B (Methodological)</em> 58 (1): 267–88.</p>
</div>
<div>
<p>Tibshirani, Robert, Jacob Bien, Jerome Friedman, Trevor Hastie, Noah Simon, Jonathan Taylor, and Ryan J Tibshirani. 2012. “Strong Rules for Discarding Predictors in Lasso-Type Problems.” <em>Journal of the Royal Statistical Society: Series B (Statistical Methodology)</em> 74 (2): 245–66.</p>
</div>
<div>
<p>Tibshirani, R., G. Walther, and T. Hastie. 2001. “Estimating the Number of Clusters in a Data Set via the Gap Statistic.” <em>Journal of the Royal Statistical Society: Series B</em> 63 (2): 411–23.</p>
</div>
<div>
<p>Visscher, Peter M., William G. Hill, and Naomi R. Wray. 2008. “Heritability in the Genomics Era-Concepts and Misconceptions.” <em>Nature Reviews Genetics</em> 9 (4): 255–66.</p>
</div>
<div>
<p>Wall, Jeffrey D, and Jonathan K Pritchard. 2003. “Haplotype Blocks and Linkage Disequilibrium in the Human Genome.” <em>Nature Reviews Genetics</em> 4 (8): 587.</p>
</div>
<div>
<p>Wang, Baohong, Mingfei Yao, Longxian Lv, Zongxin Ling, and Lanjuan Li. 2017. “The Human Microbiota in Health and Disease.” <em>Engineering</em> 3 (1): 71–82. <a href="https://doi.org/https://doi.org/10.1016/J.ENG.2017.01.008">https://doi.org/https://doi.org/10.1016/J.ENG.2017.01.008</a>.</p>
</div>
<div>
<p>Wang, Jun, and Huijue Jia. 2016. “Metagenome-Wide Association Studies: Fine-Mining the Microbiome.” <em>Nature Reviews Microbiology</em> 14 (8): 508–22.</p>
</div>
<div>
<p>Wang, Jun, Louise B. Thingholm, Jurgita Skiecevičienė, Philipp Rausch, Martin Kummen, Johannes R Hov, Frauke Degenhardt, et al. 2016. “Genome-Wide Association Analysis Identifies Variation in Vitamin d Receptor and Other Host Factors Influencing the Gut Microbiota.” <em>Nature Genetics</em>.</p>
</div>
<div>
<p>Ward, J. H. 1963. “Hierarchical Grouping to Optimize an Objective Function.” <em>Journal of the American Statistical Association</em> 58 (301): 236–44.</p>
</div>
<div>
<p>Weinberg, Wilhelm. 1908. “Ber Den Nachweis Der Vererbung Beim Menschen.” <em>Jahres. Wiertt. Ver. Vaterl. Natkd.</em> 64: 369–82.</p>
</div>
<div>
<p>Weir, Bruce S, Lon R Cardon, Amy D Anderson, Dahlia M Nielsen, and William G Hill. 2005. “Measures of Human Population Structure Show Heterogeneity Among Genomic Regions.” <em>Genome Research</em> 15 (11): 1468–76.</p>
</div>
<div>
<p>Weir, Bruce S, and C Cockerham. 1996. “Genetic Data Analysis Ii: Methods for Discrete Population Genetic Data. Sinauer Assoc.” <em>Inc., Sunderland, MA, USA</em>.</p>
</div>
<div>
<p>Weir, Bruce S, and others. 1990. <em>Genetic Data Analysis. Methods for Discrete Population Genetic Data.</em> Sinauer Associates, Inc. Publishers.</p>
</div>
<div>
<p>Wetterstrand, KA. 2016. “DNA Sequencing Costs: Data from the Nhgri Genome Sequencing Program (Gsp).” <a href="www.genome.gov/sequencingcostsdata">www.genome.gov/sequencingcostsdata</a>.</p>
</div>
<div>
<p>Williams, J. W. J. 1964. “Algorithm 232: Heapsort.” <em>Communications of the ACM</em> 7 (6): 347â348.</p>
</div>
<div>
<p>Wood, Simon N. 2006. <em>Generalized Additive Models: An Introduction with R</em>. crcpress. <a href="https://www.crcpress.com/Generalized-Additive-Models-An-Introduction-with-R/Wood/p/book/9781584884743">https://www.crcpress.com/Generalized-Additive-Models-An-Introduction-with-R/Wood/p/book/9781584884743</a>.</p>
</div>
<div>
<p>Woodrow, J. C., and C. J Eastmond. 1978. “HLA B27 and the Genetics of Ankylosing Spondylitis.” <em>Annals of the Rheumatic Diseases</em> 37 (6): 504–9.</p>
</div>
<div>
<p>Wooley, John C, Adam Godzik, and Iddo Friedberg. 2010. “A Primer on Metagenomics.” <em>PLoS Computational Biology</em> 6 (2): e1000667.</p>
</div>
<div>
<p>Wright, Alan F, and Nicholas D Hastie. 2001. “Complex Genetic Diseases: Controversy over the Croesus Code.” <em>Genome Biology</em> 2 (8): comment2007–1.</p>
</div>
<div>
<p>Wright, Sewall. 1921. “Correlation and Causation.” <em>Journal of Agricultural Research</em> 20 (7): 557–85.</p>
</div>
<div>
<p>———. 1929. “The Evolution of Dominance.” <em>The American Naturalist</em> 63 (689): 556–61.</p>
</div>
<div>
<p>WTCCC. 2007. “Genome-Wide Association Study of 14,000 Cases of Seven Common Diseases and 3,000 Shared Controls.” <em>Nature</em> 447 (7145): 661–78.</p>
</div>
<div>
<p>Wu, M. C., P. Kraft, M. P. Epstein, D. M. Taylor, S. J. Chanock, D. J. Hunter, and X. Lin. 2010. “Powerful SNP-Set Analysis for Case-Control Genome-Wide Association Studies.” <em>American Journal of Human Genetics</em> 86 (6): 929–42.</p>
</div>
<div>
<p>Wu, M. C., S. Lee, T. Cai, Y. Li, M. Boehnke, and X. Lin. 2011. “Rare-Variant Association Testing for Sequencing Data with the Sequence Kernel Association Test.” <em>American Journal of Human Genetics</em> 89 (1): 82–93.</p>
</div>
<div>
<p>Yang, Jian, Beben Benyamin, Brian P McEvoy, Scott Gordon, Anjali K Henders, Dale R Nyholt, Pamela A Madden, et al. 2010. “Common Snps Explain a Large Proportion of the Heritability for Human Height.” <em>Nature Genetics</em> 42 (7): 565.</p>
</div>
<div>
<p>Yi, Hui, Patrick Breheny, Netsanet Imam, Yongmei Liu, and Ina Hoeschele. 2015. “Penalized Multimarker Vs. Single-Marker Regression Methods for Genome-Wide Association Studies of Quantitative Traits.” <em>Genetics</em> 199 (1): 205–22.</p>
</div>
<div>
<p>Yoo, Yun Joo, Lei Sun, Julia G. Poirier, Andrew D. Paterson, and Shelley B. Bull. 2016. “Multiple Linear Combination (MLC) Regression Tests for Common Variants Adapted to Linkage Disequilibrium Structure: Yoo et Al.” <em>Genetic Epidemiology</em> 41. <a href="https://doi.org/10.1002/gepi.22024">https://doi.org/10.1002/gepi.22024</a>.</p>
</div>
<div>
<p>Yuan, Ming, and Yi Lin. 2006. “Model Selection and Estimation in Regression with Grouped Variables.” <em>JOURNAL OF THE ROYAL STATISTICAL SOCIETY, SERIES B</em> 68: 49–67.</p>
</div>
<div>
<p>Zhu, Zhihong, Zhili Zheng, Futao Zhang, Yang Wu, Maciej Trzaskowski, Robert Maier, Matthew R Robinson, et al. 2018. “Causal Associations Between Risk Factors and Common Diseases Inferred from Gwas Summary Data.” <em>Nature Communications</em> 9 (1): 224.</p>
</div>
<div>
<p>Zondervan, Krina T, and Lon R Cardon. 2004. “The Complex Interplay Among Factors That Influence Allelic Association.” <em>Nature Reviews Genetics</em> 5 (2): 89.</p>
</div>
</div>
</div>























































            </section>

          </div>
        </div>
      </div>
<a href="Bspline.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": null,
"history": {
"link": null,
"text": null
},
"download": ["book.epub", "book.pdf"],
"toc": {
"collapse": "subsection",
"scroll_highlight": true
},
"toolbar": {
"position": "fixed"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
